<!DOCTYPE html>
<html class="no-js" lang="en"> <head><meta charset="utf-8"/><meta content="width=device-width,initial-scale=1" name="viewport"/><meta content="MiXCR documentation portal" name="description"/><meta content="MiLaboratories Inc" name="author"/><link href="https://docs.milaboratories.com/mixcr/guides/generic-umi-bcr/" rel="canonical"/><link href="../../../assets/favicon.png" rel="icon"/><meta content="mkdocs-1.4.0, mkdocs-material-8.3.9" name="generator"/><title>Multiplex BCR UMI cDNA libraries - Docs @ MiLaboratories</title><link href="../../../assets/stylesheets/main.1d29e8d0.min.css" rel="stylesheet"/><link href="../../../assets/stylesheets/palette.cbb835fc.min.css" rel="stylesheet"/><link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/><link href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,300i,400,400i,700,700i%7CIBM+Plex+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/><style>:root{--md-text-font:"IBM Plex Sans";--md-code-font:"IBM Plex Mono"}</style><link href="../../../css/extra.css" rel="stylesheet"/><script>__md_scope=new URL("../../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-1KLG43HZP1"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&gtag("event","search",{search_term:this.value})}),"undefined"!=typeof location$&&location$.subscribe(function(e){gtag("config","G-1KLG43HZP1",{page_path:e.pathname})})})</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-1KLG43HZP1"></script><link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gdesc-inner { font-size: 0.75rem; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                </style><script src="../../../assets/javascripts/glightbox.min.js"></script></head> <body data-md-color-accent="" data-md-color-primary="" data-md-color-scheme="default" dir="ltr"> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/> <input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/> <label class="md-overlay" for="__drawer"></label> <div data-md-component="skip"> <a class="md-skip" href="#multiplex-bcr-umi-cdna-libraries"> Skip to content </a> </div> <div data-md-component="announce"> </div> <div data-md-component="outdated" hidden=""> <aside class="md-banner md-banner--warning"> </aside> </div> <header class="md-header" data-md-component="header"> <nav aria-label="Header" class="md-header__inner md-grid"> <a aria-label="Docs @ MiLaboratories" class="md-header__button md-logo" data-md-component="logo" href="../../.." title="Docs @ MiLaboratories"> <img alt="logo" src="../../../assets/logo.svg"/> </a> <label class="md-header__button md-icon" for="__drawer"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg> </label> <div class="md-header__title" data-md-component="header-title"> <div class="md-header__ellipsis"> <div class="md-header__topic"> <span class="md-ellipsis"> Docs @ MiLaboratories </span> </div> <div class="md-header__topic" data-md-component="header-topic"> <span class="md-ellipsis"> Multiplex BCR UMI cDNA libraries </span> </div> </div> </div> <form class="md-header__option" data-md-component="palette"> <input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/> <label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to dark mode"> <svg fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <path clip-rule="evenodd" d="M18.3578 4.32112C18.2104 4.39482 18.2104 4.60518 18.3578 4.67889L18.9404 4.97019C18.9791 4.98954 19.0105 5.02092 19.0298 5.05963L19.3211 5.64223C19.3948 5.78964 19.6052 5.78964 19.6789 5.64223L19.9702 5.05963C19.9895 5.02092 20.0209 4.98954 20.0596 4.97019L20.6422 4.67889C20.7896 4.60518 20.7896 4.39482 20.6422 4.32112L20.0596 4.02982C20.0209 4.01046 19.9895 3.97908 19.9702 3.94037L19.6789 3.35777C19.6052 3.21036 19.3948 3.21036 19.3211 3.35777L19.0298 3.94037C19.0105 3.97908 18.9791 4.01046 18.9404 4.02982L18.3578 4.32112ZM18.2083 16.2099C17.5036 16.3991 16.7631 16.5 16 16.5C11.3056 16.5 7.5 12.6944 7.5 8C7.5 7.23686 7.60088 6.4964 7.79006 5.79169C5.80401 7.14138 4.5 9.41947 4.5 12C4.5 16.1421 7.85786 19.5 12 19.5C14.5805 19.5 16.8586 18.196 18.2083 16.2099ZM19.1573 14.2492C18.2087 14.7294 17.136 15 16 15C12.134 15 9 11.866 9 8C9 6.86405 9.27058 5.79129 9.75081 4.84267L9.77821 4.78912C9.83867 4.6722 9.90234 4.55719 9.96908 4.44423C10.0299 4.34128 10.0933 4.24002 10.1592 4.14055C10.4192 3.74773 10.1191 3.18416 9.66405 3.30612C9.48653 3.3537 9.31118 3.40657 9.13817 3.46455C9.01891 3.50452 8.90077 3.54692 8.7838 3.59169C8.76089 3.60045 8.73803 3.60931 8.71521 3.61826C5.3692 4.93057 3 8.18861 3 12C3 16.9706 7.02944 21 12 21C15.8114 21 19.0694 18.6308 20.3817 15.2848C20.3907 15.262 20.3995 15.2391 20.4083 15.2162C20.4531 15.0992 20.4955 14.9811 20.5354 14.8618C20.5934 14.6888 20.6463 14.5135 20.6939 14.3359C20.8158 13.8809 20.2523 13.5808 19.8595 13.8408C19.76 13.9067 19.6587 13.9701 19.5558 14.0309C19.4428 14.0977 19.3278 14.1613 19.2109 14.2218L19.1573 14.2492ZM16.4602 7.40052L15.7321 5.5803C15.6483 5.37076 15.3517 5.37076 15.2679 5.5803L14.5398 7.40052C14.5144 7.46405 14.464 7.51438 14.4005 7.53979L12.5803 8.26788C12.3708 8.3517 12.3708 8.64831 12.5803 8.73212L14.4005 9.46021C14.464 9.48562 14.5144 9.53596 14.5398 9.59948L15.2679 11.4197C15.3517 11.6292 15.6483 11.6292 15.7321 11.4197L16.4602 9.59948C16.4856 9.53596 16.536 9.48562 16.5995 9.46021L18.4197 8.73212C18.6292 8.64831 18.6292 8.3517 18.4197 8.26788L16.5995 7.53979C16.536 7.51438 16.4856 7.46405 16.4602 7.40052Z" fill="white" fill-rule="evenodd"></path> </svg> </label> <input aria-label="Switch to light mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="" data-md-color-scheme="slate" id="__palette_2" name="__palette" type="radio"/> <label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to light mode"> <svg fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <path clip-rule="evenodd" d="M12.75 3V0H11.25V3H12.75ZM12.75 24V21H11.25V24H12.75ZM3 12.75H0V11.25H3V12.75ZM21 12.75H24V11.25H21V12.75ZM12 16.5C14.4853 16.5 16.5 14.4853 16.5 12C16.5 9.51472 14.4853 7.5 12 7.5C9.51472 7.5 7.5 9.51472 7.5 12C7.5 14.4853 9.51472 16.5 12 16.5ZM12 18C15.3137 18 18 15.3137 18 12C18 8.68629 15.3137 6 12 6C8.68629 6 6 8.68629 6 12C6 15.3137 8.68629 18 12 18ZM21.0156 4.04506L18.8943 6.16638L17.8336 5.10572L19.955 2.9844L21.0156 4.04506ZM5.10571 17.8336L6.16637 18.8943L4.04505 21.0156L2.98439 19.955L5.10571 17.8336ZM2.98439 4.04506L5.10571 6.16638L6.16637 5.10572L4.04505 2.9844L2.98439 4.04506ZM21.0156 19.955L19.955 21.0156L17.8336 18.8943L18.8943 17.8336L21.0156 19.955Z" fill="white" fill-rule="evenodd"></path> </svg> </label> </form> <label class="md-header__button md-icon" for="__search"> <svg class="magnify" fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <circle cx="11" cy="11" r="7.25" stroke="white" stroke-width="1.5"></circle> <path d="M16 16L21.5 21.5" stroke="white" stroke-width="1.5"></path> </svg> </label> <div class="md-search" data-md-component="search" role="dialog"> <label class="md-search__overlay" for="__search"></label> <div class="md-search__inner" role="search"> <form class="md-search__form" name="search"> <input aria-label="Search by name or category" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search by name or category" required="" spellcheck="false" type="text"/> <label class="md-search__icon md-icon" for="__search"> <svg class="magnify" fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <circle cx="11" cy="11" r="7.25" stroke="white" stroke-width="1.5"></circle> <path d="M16 16L21.5 21.5" stroke="white" stroke-width="1.5"></path> </svg> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg> </label> <nav aria-label="Search" class="md-search__options"> <a aria-label="Share" class="md-search__icon md-icon" data-clipboard="" data-clipboard-text="" data-md-component="search-share" href="javascript:void(0)" tabindex="-1"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"></path></svg> </a> <button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" type="reset"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg> </button> </nav> <div class="md-search__suggest" data-md-component="search-suggest"></div> </form> <div class="md-search__output"> <div class="md-search__scrollwrap" data-md-scrollfix=""> <div class="md-search-result" data-md-component="search-result"> <div class="md-search-result__meta"> Initializing search </div> <ol class="md-search-result__list"></ol> </div> </div> </div> </div> </div> <div class="md-header__source"> <a class="md-source" href="https://github.com/milaboratory/docs" title="Go to repository"> <div class="md-source__icon md-icon"> <svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg> </div> <div class="md-source__repository"> milaboratory/docs </div> </a> </div> </nav> </header> <div class="md-container" data-md-component="container"> <nav aria-label="Tabs" class="md-tabs" data-md-component="tabs"> <div class="md-tabs__inner md-grid"> <ul class="md-tabs__list"> <li class="md-tabs__item"> <a class="md-tabs__link" href="../../.."> Home </a> </li> <li class="md-tabs__item"> <a class="md-tabs__link" href="../../getting-started/installation/"> Getting Started </a> </li> <li class="md-tabs__item"> <a class="md-tabs__link" href="../milaboratories-human-tcr-rna-multi/"> Guides </a> </li> <li class="md-tabs__item"> <a class="md-tabs__link" href="../../reference/mixcr-analyze/"> Reference </a> </li> </ul> </div> </nav> <main class="md-main" data-md-component="main"> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation"> <div class="md-sidebar__scrollwrap"> <div class="md-sidebar__inner"> <!-- Determine class according to configuration --> <!-- Ugly hack --> <!-- Main navigation --> <nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted" data-md-level="0" data-test="1"> <!-- Site title --> <label class="md-nav__title" for="__drawer"> <a aria-label="Docs @ MiLaboratories" class="md-nav__button md-logo" data-md-component="logo" href="../../.." title="Docs @ MiLaboratories"> <img alt="logo" src="../../../assets/logo.svg"/> </a> Docs @ MiLaboratories </label> <!-- Repository information --> <div class="md-nav__source"> <a class="md-source" href="https://github.com/milaboratory/docs" title="Go to repository"> <div class="md-source__icon md-icon"> <svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg> </div> <div class="md-source__repository"> milaboratory/docs </div> </a> </div> <!-- Render item list --> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../.."> Home </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" id="__nav_2" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="1" for="__nav_2"> Getting Started <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Getting Started" class="md-nav" data-md-level="1"> <label class="md-nav__title" for="__nav_2"> <span class="md-nav__icon md-icon"></span> Getting Started </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/installation/"> Installation </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/docker/"> Using with Docker </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/milm/"> Obtaining and using license key </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/overview/"> MiXCR Overview </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/license/"> License </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" id="__nav_3" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="1" for="__nav_3"> Guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Guides" class="md-nav" data-md-level="1"> <label class="md-nav__title" for="__nav_3"> <span class="md-nav__icon md-icon"></span> Guides </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_1" id="__nav_3_1" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_3_1"> Kits &amp; Protocols <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Kits &amp; Protocols" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_3_1"> <span class="md-nav__icon md-icon"></span> Kits &amp; Protocols </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../milaboratories-human-tcr-rna-multi/"> MiLaboratories Human TCR RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../takara-hsa-bcr/"> SMARTer Human BCR IgG IgM H/K/L Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../takara-mmu-tcr/"> SMARTer Mouse TCR a/b Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../ampliseq-tcr-lr/"> Illumina Ampliseq TCR LR / iRepertoire </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../nebnext-bcr/"> NEBNext Immune Sequencing </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../qiaseq-tcr/"> QIAseq Immune Repertoire TCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../abhelix-bcr/"> AbHelix BCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../biomed2-bcr/"> BioMed2 </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_2" id="__nav_3_2" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_3_2"> Analysis cases <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Analysis cases" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_3_2"> <span class="md-nav__icon md-icon"></span> Analysis cases </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../generic-umi-tcr/"> Targeted TCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../generic-multiplex-bcr/"> Targeted BCR Multiplex libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../rnaseq/"> RNA-Seq data </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_3" id="__nav_3_3" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_3_3"> Special guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Special guides" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_3_3"> <span class="md-nav__icon md-icon"></span> Special guides </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../create-custom-library/"> Assembling custom reference library </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../external-libraries/"> Using external libraries </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" id="__nav_4" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="1" for="__nav_4"> Reference <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Reference" class="md-nav" data-md-level="1"> <label class="md-nav__title" for="__nav_4"> <span class="md-nav__icon md-icon"></span> Reference </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_1" id="__nav_4_1" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_1"> MiXCR <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="MiXCR" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_1"> <span class="md-nav__icon md-icon"></span> MiXCR </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-analyze/"> analyze </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-align/"> align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-correctAndSortTags/"> correctAndSortTags </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-assemblePartial/"> assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-extend/"> extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-assemble/"> assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-assembleContigs/"> assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-export/"> export </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportPretty/"> exportPretty </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportAirr/"> exportAirr </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportReadsForClones/"> exportReadsForClones </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportReports/"> exportReports </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportQc/"> exportQc </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-postanalysis/"> postanalysis </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportPlots/"> exportPlots </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportTables/"> exportTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportPreprocTables/"> exportPreprocTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportClonesOverlap/"> exportClonesOverlap </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-overlapScatterPlot/"> overlapScatterPlot </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" id="__nav_4_2" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_2"> Reports <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Reports" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_2"> <span class="md-nav__icon md-icon"></span> Reports </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-align/"> mixcr align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-correctAndSortTags/"> mixcr correctAndSortTags </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-assemblePartial/"> mixcr assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-extend/"> mixcr extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-assemble/"> mixcr assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-assembleContigs/"> mixcr assembleContigs </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_3" id="__nav_4_3" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_3"> RepSeq.io <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="RepSeq.io" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_3"> <span class="md-nav__icon md-icon"></span> RepSeq.io </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-fasta/"> fasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-compile/"> compile </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-debug/"> debug </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-filter/"> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-format/"> format </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-fromFasta/"> fromFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-fromPaddedFasta/"> fromPaddedFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-inferPoints/"> inferPoints </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-merge/"> merge </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-stat/"> stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-tsv/"> tsv </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_4" id="__nav_4_4" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_4"> MiTool <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="MiTool" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_4"> <span class="md-nav__icon md-icon"></span> MiTool </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-parse/"> parse </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-filter/"> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-tag-stat/"> tag-stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-export-fastq/"> export-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-correct/"> correct </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-consensus/"> consensus </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-process-fastq/"> process-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-format-fastq/"> format-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-compress-sequence-set/"> compress-sequence-set </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_5" id="__nav_4_5" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_5"> Specs <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Specs" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_5"> <span class="md-nav__icon md-icon"></span> Specs </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/ref-tag-pattern/"> Tag Pattern </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/ref-gene-features/"> Gene features and reference points </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/ref-translation-rules/"> Translation rules </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/ref-mutations-encoding/"> Mutations encoding </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-json-format/"> RepSeq.io JSON format </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc"> <div class="md-sidebar__scrollwrap"> <div class="md-sidebar__inner"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav aria-label="Table of contents" class="md-nav md-nav--secondary hello"> <label class="md-nav__title" for="__toc"> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix=""> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#experiment-setting"> Experiment setting </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#one-command-solution"> One command Solution </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#under-the-hood-of-mixcr-analyze-pipeline"> Under the hood of mixcr analyze pipeline </a> <!-- Table of contents list --> <nav aria-label="Under the hood of mixcr analyze pipeline" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#align"> align </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#correctandsorttags"> correctAndSortTags </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#assemble"> assemble </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#exportclones"> exportClones </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#quality-control"> Quality control </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#downstream-analysis"> Downstream analysis </a> <!-- Table of contents list --> <nav aria-label="Downstream analysis" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#individual-postanalysis"> Individual postanalysis </a> <!-- Table of contents list --> <nav aria-label="Individual postanalysis" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#graphical-output"> Graphical output </a> <!-- Table of contents list --> <nav aria-label="Graphical output" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#diversity"> Diversity </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#v-usage"> V usage </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#overlap-postanalysis"> Overlap postanalysis </a> <!-- Table of contents list --> <nav aria-label="Overlap postanalysis" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#two-samples-overlap"> Two samples overlap </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#all-vs-all-overlap"> All-vs-All overlap </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> </ul> </nav> </div> </div> </div> <div class="md-content" data-md-component="content"> <article class="md-content__inner md-typeset"> <a class="md-content__button md-icon" href="https://github.com/milaboratory/docs/edit/main/docs/mixcr/guides/generic-umi-bcr.md" title="Edit this page"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"></path></svg> </a> <h1 id="multiplex-bcr-umi-cdna-libraries">Multiplex BCR UMI cDNA libraries</h1> <p>For this tutorial we will use the data published in the following article:</p> <p>Chen JS, Chow RD, Song E, Mao T, Israelow B, Kamath K, Bozekowski J, Haynes WA, Filler RB, Menasche BL, Wei J, Alfajaro MM, Song W, Peng L, Carter L, Weinstein JS, Gowthaman U, Chen S, Craft J, Shon JC, Iwasaki A, Wilen CB, Eisenbarth SC. <em>High-affinity, neutralizing antibodies to SARS-CoV-2 can be made without T follicular helper cells.</em> Sci Immunol. 2022 Feb 4;7(68):eabl5652. doi: 10.1126/sciimmunol.abl5652. Epub 2022 Feb 4. PMID: 34914544; PMCID: PMC8977051.</p> <h2 id="experiment-setting">Experiment setting</h2> <p>The experiment includes 8 mice that express angiotensin-converting enzyme 2 (ACE2) in epithelial cells thus vulnerable to SARS-CoV-2. 4 K18-hACE2 Bcl6fl/fl mice and 4 Bcl6fl/flCd4Cre (T~FH~ cell deficiency model) mice. Mice were infected intranasally with SARS-CoV-2 (isolate USA-WA1/2020) and 14 days post-infection S-specific plasmablasts were isolated. RNA from spike-specific B cells was isolated using the RNeasy Plus Micro Kit (Qiagen) following the manufacturer’s instructions. BCR libraries were prepared using the NEBNext Single Cell/Low Input cDNA Synthesis &amp; Amplification Module (New England Biolabs (NEB), E6421) and NEBNext Immune Sequencing Kit (NEB, E6330), with additional reagents provided by NEB to integrate the two kits. High-quality RNA (1 to 20 ng) with RNA integrity number ≥ 8 was used as input. Libraries were analyzed by Bioanalyzer High Sensitivity DNA assay, pooled in equal amounts with PhiX spike-in, and sequenced on an Illumina MiSeq using the V3 kit, with 325 base pairs (bp) for read 1 and 275 bp for read 2.</p> <figure> <p><a class="glightbox" data-desc-position="bottom" data-description="" data-height="auto" data-title="" data-width="100%" href="../generic-multiplex-bcr/PRJNA699402-library-structure.svg"><img alt="PRJNA699402-library-structure.svg" src="../generic-multiplex-bcr/PRJNA699402-library-structure.svg"/></a></p> </figure> <p>All data may be downloaded directly from SRA using e.g. <a href="https://sra-explorer.info">SRA Explorer</a> and PRJNA780318 project id :</p> <details class="tip"> <summary>Use this script to download the full data set with the proper filenames for the tutorial:</summary> <div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/env bash</span>
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/022/SRR16943422/SRR16943422_1.fastq.gz -o mouse_3_Cd4Cre_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/022/SRR16943422/SRR16943422_2.fastq.gz -o mouse_3_Cd4Cre_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/025/SRR16943425/SRR16943425_1.fastq.gz -o mouse_4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/025/SRR16943425/SRR16943425_2.fastq.gz -o mouse_4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/028/SRR16943428/SRR16943428_1.fastq.gz -o mouse_1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/028/SRR16943428/SRR16943428_2.fastq.gz -o mouse_1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/021/SRR16943421/SRR16943421_1.fastq.gz -o mouse_4_Cd4Cre_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/021/SRR16943421/SRR16943421_2.fastq.gz -o mouse_4_Cd4Cre_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/023/SRR16943423/SRR16943423_1.fastq.gz -o mouse_2_Cd4Cre_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/023/SRR16943423/SRR16943423_2.fastq.gz -o mouse_2_Cd4Cre_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/027/SRR16943427/SRR16943427_1.fastq.gz -o mouse_2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/027/SRR16943427/SRR16943427_2.fastq.gz -o mouse_2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/024/SRR16943424/SRR16943424_1.fastq.gz -o mouse_1_Cd4Cre_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/024/SRR16943424/SRR16943424_2.fastq.gz -o mouse_1_Cd4Cre_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/026/SRR16943426/SRR16943426_1.fastq.gz -o mouse_3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR169/026/SRR16943426/SRR16943426_2.fastq.gz -o mouse_3_R2.fastq.gz
</code></pre></div> </details> <h2 id="one-command-solution">One command Solution</h2> <p>The easiest way to obtain clonotype tables for this type of data is to use a universal <a href="../../reference/mixcr-analyze/"><code>mixcr analyze</code></a> command.</p> <p>The exact command for a single sample you can see bellow:</p> <div class="highlight"><pre><span></span><code><span class="err">'</span>&gt; mixcr analyze amplicon <span class="se">\</span>
    --species mmu <span class="se">\</span>
    --starting-material rna <span class="se">\</span>
    --receptor-type bcr <span class="se">\</span>
    --5-end no-v-primers <span class="se">\</span>
    --3-end c-primers <span class="se">\</span>
    --adapters adapters-present <span class="se">\</span>
    --assemble <span class="s2">"-OassemblingFeatures={CDR1Begin:FR4End} -OseparateByC=true"</span> <span class="se">\</span>
    mouse_3_Cd4Cre_R1.fastq.gz <span class="se">\</span>
    mouse_3_Cd4Cre_R2.fastq.gz <span class="se">\</span>
    mouse_3_Cd4Cre
</code></pre></div> <p>The meaning of these options is the following.</p> <dl> <dt><code>--species</code></dt> <dd>is set to <code>mmu</code> for <em>Mus Musculus</em></dd> <dt><code>--starting-material</code></dt> <dd>is set to <code>rna</code> and corresponds to <code>VTranscriptWithout5UTRWithP</code> alignment feature for V-gene (see <a href="../../reference/ref-gene-features/">Gene features and anchor points</a> for details)</dd> <dt><code>--receptor-type</code></dt> <dd><code>bcr</code>. It affects the choice of alignment algorithms. MiXCR uses a specific set of algorithms for BCR data.</dd> <dt><code>--5-end</code></dt> <dd>is set to <code>no-v-primers</code> because NEBNext Immune Sequencing Kit uses a 5'RACE technology that does not require V gene region primers . This leads to a global alignment algorithms to align the left bound of V gene.</dd> <dt><code>--3-end-primers</code></dt> <dd>is set to <code>c-primers</code>. Here, C primers were used in cDNA synthesis and these primers are isotype specific. This choice leads to a global alignment algorithms to align the right bound of J gene and a local alignment algorithm for the left bound of C-gene.</dd> <dt><code>--adapers</code></dt> <dd><code>adapters-present</code> because we still have C-primer sequences in our data.</dd> <dt><code>--assemble</code></dt> <dd><code>"-OassemblingFeatures={CDR1Begin:FR4End} -OseparateByC=true"</code>. Here we pass two extra arguments for <a href="../../reference/mixcr-assemble/"><code>mixcr assemble</code></a> step of the pipeline. First we extend the assembling feature to start from <code>CDR1</code>. That is because this is BCR data, where hypermutations occur throughout V gene, and we want to capture as much as we can. Second, we use <code>-OseparateByC=true</code> option to separate clones with the same assembling feature sequence but different C genes, which is essential for isotype identification.</dd> <dt><code>mouse_3_Cd4Cre_R1.fastq.gz mouse_3_Cd4Cre_R2.fastq.gz mouse_3_Cd4Cre</code></dt> <dd>Finally, we provide the names of input files and an output prefix:</dd> </dl> <p>Now, since we have multiple files ist easier to process them all together instead of running the same command multiple times. One of the ways to achieve it is to use <a href="https://www.gnu.org/software/parallel/">GNU Parallel</a>:</p> <div class="highlight"><pre><span></span><code>&gt; fastq/*R1* <span class="p">|</span> 
  parallel -j4 <span class="se">\</span>
  <span class="s1">'mixcr analyze amplicon \</span>
<span class="s1">    --species mmu \</span>
<span class="s1">    --starting-material rna \</span>
<span class="s1">    --receptor-type bcr \</span>
<span class="s1">    --5-end no-v-primers \</span>
<span class="s1">    --3-end c-primers \</span>
<span class="s1">    --adapters adapters-present \</span>
<span class="s1">    --umi-pattern "^(R1:*)\^(UMI:N{"17"})(R2:*)" \</span>
<span class="s1">    --assemble "-OassemblingFeatures={FR1Begin:FR4End} -OseparateByC=true"\</span>
<span class="s1">    {} \</span>
<span class="s1">    {=s:R1:R2:=} \</span>
<span class="s1">    {=s:.*/:results/:;s:_R1.*::=}'</span>
</code></pre></div> <h2 id="under-the-hood-of-mixcr-analyze-pipeline">Under the hood of <code>mixcr analyze</code> pipeline</h2> <p>Under the hood, <code>mixcr analyze amplicon</code> command that we use above actually executes the following pipeline of MiXCR actions:</p> <h4 id="align"><code>align</code></h4> <p><a href="../../reference/mixcr-align/">Performs</a>:</p> <ul> <li>alignment of raw sequencing reads against reference database of V-, D-, J- and C- gene segments</li> <li>pattern matching of tag pattern sequence and extraction of barcodes</li> </ul> <div class="highlight"><pre><span></span><code> &gt; mixcr align <span class="se">\</span>
    --species mmu <span class="se">\</span>
    --report result/Multi_TRA_10ng_3.report <span class="se">\</span>
    --json-report result/Multi_TRA_10ng_3.report.json <span class="se">\</span>
    -OvParameters.geneFeatureToAlign<span class="o">={</span>CDR1Begin:VEnd<span class="o">}</span>+<span class="o">{</span>VEnd:VEnd<span class="o">(</span>-20<span class="o">)}</span> <span class="se">\</span>
    -OvParameters.parameters.floatingLeftBound<span class="o">=</span><span class="nb">false</span> <span class="se">\</span>
    -OjParameters.parameters.floatingRightBound<span class="o">=</span><span class="nb">false</span> <span class="se">\</span>
    -OcParameters.parameters.floatingRightBound<span class="o">=</span><span class="nb">true</span> <span class="se">\</span>
    fastq/mouse_3_Cd4Cre_R1.fastq.gz <span class="se">\</span>
    fastq/mouse_3_Cd4Cre_R2.fastq.gz <span class="se">\</span>
    results/mouse_3_Cd4Cre.vdjca
</code></pre></div> <p>Options <code>--report</code> and <code>--json-report</code> are specified here explicitly. Since we start from RNA data we use <code>VTranscriptWithout5UTRWithP</code> for the alignment of V segments (see <a href="../../reference/ref-gene-features/">Gene features and anchor points</a>.</p> <dl> <dt><code>-OvParameters.parameters.floatingLeftBound=false -OjParameters.parameters floatingRightBound=false -OcParameters.parameters.floatingRightBound=true</code></dt> <dd>These options determine global vs local alignment algorithm on the bounds of gene segments. As have been mentioned above we will use global alignment on the left bound of V gene and right bound of J gene. The alignment on the right bound of C gene will be local due to the presence of primer sequence.</dd> </dl> <p>This step utilizes all available CPUs and scales perfectly. When there are a lot of CPUs, the only limiting factor is the speed of disk I/O. To limit the number of used CPUs one can pass <code>--threads N</code> option.</p> <h4 id="correctandsorttags"><code>correctAndSortTags</code></h4> <p><a href="../../reference/mixcr-correctAndSortTags/">Corrects</a> sequencing and PCR errors <em>inside</em> barcode sequences. This step does extremely important job by correcting artificial diversity caused by errors in barcodes. In the considered example project it corrects only sequences of UMIs.</p> <div class="highlight"><pre><span></span><code>&gt; mixcr correctAndSortTags <span class="se">\</span>
    --report results/mouse_3_Cd4Cre.report <span class="se">\</span>
    --json-report results/mouse_3_Cd4Cre.json <span class="se">\</span>
    mouse_3_Cd4Cre.vdjca <span class="se">\</span>
    mouse_3_Cd4Cre.corrected.vdjca
</code></pre></div> <p>Options <code>--report</code> and <code>--json-report</code> are specified here explicitly so that the report files will be appended with the barcode correction report.</p> <h4 id="assemble"><code>assemble</code></h4> <p><a href="../../reference/mixcr-assemble/">Assembles</a> clonotypes and applies several layers of errors correction:</p> <ul> <li>assembly consensus CDR3 sequence</li> <li>quality-awared correction for sequencing errors</li> <li>clustering to correct for PCR errors</li> </ul> <div class="highlight"><pre><span></span><code>&gt; mixcr assemble <span class="se">\</span>
    --report results/mouse_3_Cd4Cre.report <span class="se">\</span>
    --json-report results/mouse_3_Cd4Cre.report.json <span class="se">\</span>
    -OassemblingFeatures<span class="o">={</span>CDR1Begin:FR4End<span class="o">}</span> <span class="se">\</span>
    -OseparateByC<span class="o">=</span><span class="nb">true</span> <span class="se">\</span>
    mouse_3_Cd4Cre.corrected.vdjca <span class="se">\</span>
    mouse_3_Cd4Cre.clns
</code></pre></div> <p>Options <code>--report</code> and <code>--json-report</code> are specified here explicitly so that the report files will be appended with assembly report.</p> <h4 id="exportclones"><code>exportClones</code></h4> <p>Finally, to <a href="../../reference/mixcr-export/#clonotype-tables">export</a> clonotype tables in tabular form <code>exportClones</code> is used. By default <code>mixcr analyze</code> will export all availible chains into separate files which is equivalent to running a set of the following commands :</p> <div class="highlight"><pre><span></span><code>&gt; mixcr exportClones <span class="se">\</span>
    -p full <span class="se">\</span>
    -c IGH <span class="se">\</span>
    mouse_3_Cd4Cre.clns <span class="se">\</span>
    mouse_3_Cd4Cre.clonotypes.IGH.tsv

&gt; mixcr exportClones <span class="se">\</span>
    -p full <span class="se">\</span>
    -c IGK <span class="se">\</span>
    mouse_3_Cd4Cre.clns <span class="se">\</span>
    mouse_3_Cd4Cre.clonotypes.IGK.tsv

&gt; mixcr exportClones <span class="se">\</span>
    -p full <span class="se">\</span>
    -c IGL <span class="se">\</span>
    mouse_3_Cd4Cre.clns <span class="se">\</span>
    mouse_3_Cd4Cre.clonotypes.IGL.tsv
</code></pre></div> <p>Here <code>-p full</code> is a shorthand for the full preset of common export columns.</p> <h2 id="quality-control">Quality control</h2> <p>Now when the upstream analysis is finished we can move on to quality control. First lets look at the alignment report plot.</p> <div class="highlight"><pre><span></span><code><span class="c1"># obtain alignment quality control</span>
&gt; mixcr exportQc align <span class="se">\</span>
    result/*.vdjca <span class="se">\</span>
    alignQc.pdf
</code></pre></div> <figure> <p><a class="glightbox" data-desc-position="bottom" data-description="" data-height="auto" data-title="" data-width="100%" href="alignQc.svg"><img alt="alignQc.svg" src="alignQc.svg"/></a></p> </figure> <p>We see that all samples have a very high score of successfully aligned reads. No significant issues present.</p> <p>Next, lets examine chain usage distribution</p> <figure> <p><a class="glightbox" data-desc-position="bottom" data-description="" data-height="auto" data-title="" data-width="100%" href="chainUsage.svg"><img alt="chainUsage.svg" src="chainUsage.svg"/></a></p> </figure> <p>This plot tells us that all samples mostly consist of IGH and IGK with relatively small numbers of IGL, which is pretty much what we would expect. Thus, quality reports look good and we can move on to downstream analysis.</p> <h2 id="downstream-analysis">Downstream analysis</h2> <p>There are two types of basic downstream analysis: <em>individual</em> and <em>overlap</em>. Individual computes CDR3 metrics, diversity and gene usage metrics for each dataset. Overlap computes statistical metrics of repertoires overlap. In both cases MiXCR applies appropriate sample normalization.</p> <p>To run postanalysis routines we need to prepare a metadata file in a .tsv or .csv form. Metadata must contain a <code>sample</code> column which will be used to match metadata with cloneset files. Bellow you can find a metadata table for our samples.</p> <details class="note"> <summary>Metadata</summary> <table> <thead> <tr> <th>sample</th> <th>genotype</th> </tr> </thead> <tbody> <tr> <td>mouse_1_Cd4Cre</td> <td>Bcl6fl/flCd4Cre</td> </tr> <tr> <td>mouse_1</td> <td>Bcl6fl/fl</td> </tr> <tr> <td>mouse_2_Cd4Cre</td> <td>Bcl6fl/flCd4Cre</td> </tr> <tr> <td>mouse_2</td> <td>Bcl6fl/fl</td> </tr> <tr> <td>mouse_3_Cd4Cre</td> <td>Bcl6fl/flCd4Cre</td> </tr> <tr> <td>mouse_3</td> <td>Bcl6fl/fl</td> </tr> <tr> <td>mouse_4_Cd4Cre</td> <td>Bcl6fl/flCd4Cre</td> </tr> <tr> <td>mouse_4</td> <td>Bcl6fl/fl</td> </tr> </tbody> </table> </details> <h3 id="individual-postanalysis">Individual postanalysis</h3> <p>To compute a set of individual metrics we run the following command:</p> <div class="highlight"><pre><span></span><code>&gt; mixcr postanalysis individual <span class="se">\</span>
    --metadata metadata.tsv <span class="se">\</span>
    --default-downsampling count-umi-auto <span class="se">\</span>
    --default-weight-function umi <span class="se">\</span>
    --only-productive <span class="se">\</span>
    --tables postanalysis/pa.i.tsv <span class="se">\</span>
    --chains IG <span class="se">\</span>
    --preproc-tables postanalysis/preproc.i.tsv <span class="se">\</span>
    results/*.clns <span class="se">\</span>
    postanalysis/individual.json.gz
</code></pre></div> <p>The meaning of specified options is the following:</p> <dl> <dt><code>--metadata</code></dt> <dd>specified metadata file to use</dd> <dt><code>--default-downsampling</code></dt> <dd>downsampling applied to normalize the clonesets. Without appropriate normalization it is not possible to make a statistical comparisons between datasets. In the considered case we normalize data to the same number of UMIs, and this number is computed automatically. See <a href="../../reference/mixcr-postanalysis/#downsampling">reference</a> for all downsampling options. Default downsampling may be overridden for individual metrics.</dd> <dt><code>--default-weight-function</code></dt> <dd>defines weight of each clonotype. Set <code>umi</code> because our data is barcoded and instead of using reads count we can use UMIs count for every clone.</dd> <dt><code>--only-productive</code></dt> <dd>drop clonotypes with out-of-frame CDR3 sequences or containing stop codons</dd> <dt><code>--tables</code></dt> <dd>path for postanalysis metrics in a tabular form</dd> <dt><code>--preproc-tables</code></dt> <dd>path for tabular summary of the applied downsampling and other samples preprocessing (for example filtering productive clonotypes)</dd> </dl> <p>After execution, we will have the following files:</p> <div class="highlight"><pre><span></span><code>&gt; ls postanalysis/

<span class="c1"># gzipped JSON with all results </span>
individual.json.gz

<span class="c1"># summary of applied preprocessors</span>
preproc.i.IGH.tsv
preproc.i.IGK.tsv
preproc.i.IGL.tsv

<span class="c1"># diversity tables</span>
pa.i.diversity.IGH.tsv
pa.i.diversity.IGK.tsv
pa.i.diversity.IGL.tsv

<span class="c1"># CDR3 metrics tables &amp; CDR3 properties</span>
pa.i.cdr3metrics.IGH.tsv
pa.i.cdr3metrics.IGK.tsv
pa.i.cdr3metrics.IGL.tsv


<span class="c1"># V- / J- / VJ- usage tables</span>
pa.i.vUsage.IGH.tsv
pa.i.JUsage.IGH.tsv
pa.i.VJUsage.IGK.tsv
pa.i.vUsage.IGH.tsv
pa.i.JUsage.IGK.tsv
pa.i.VJUsage.IGL.tsv
pa.i.vUsage.IGH.tsv
pa.i.JUsage.IGK.tsv
pa.i.VJUsage.IGL.tsv

<span class="c1"># Isotype usage table</span>
pa.i.IsotypeUsage.IGH.tsv
pa.i.IsotypeUsage.IGK.tsv
pa.i.IsotypeUsage.IGL.tsv


<span class="c1">#V Spectratype tables</span>
pa.i.VSpectratype.IGH.tsv
pa.i.VSpectratypeMean.IGH.tsv
pa.i.VSpectratype.IGK.tsv
pa.i.VSpectratypeMean.IGK.tsv
pa.i.VSpectratype.IGL.tsv
pa.i.VSpectratypeMean.IGL.tsv
</code></pre></div> <p>Preprocessing summary tables (e.g. <code>preproc.i.IGH.tsv</code>) contain detailed information on how downsampling was applied for each metric:</p> <table> <thead> <tr> <th>characteristic</th> <th>sample</th> <th>preprocessor</th> <th>nElementsBefore</th> <th>sumWeightBefore</th> <th>nElementsAfter</th> <th>sumWeightAfter</th> <th>preprocessor#1</th> <th>nElementsBefore#1</th> <th>sumWeightBefore#1</th> <th>nElementsAfter#1</th> <th>sumWeightAfter#1</th> <th>preprocessor#2</th> <th>nElementsBefore#2</th> <th>sumWeightBefore#2</th> <th>nElementsAfter#2</th> <th>sumWeightAfter#2</th> <th>preprocessor#3</th> <th>nElementsBefore#3</th> <th>sumWeightBefore#3</th> <th>nElementsAfter#3</th> <th>sumWeightAfter#3</th> <th>preprocessor#4</th> <th>nElementsBefore#4</th> <th>sumWeightBefore#4</th> <th>nElementsAfter#4</th> <th>sumWeightAfter#4</th> </tr> </thead> <tbody> <tr> <td>IsotypeUsage</td> <td>mouse_4_Cd4Cre.clns</td> <td>Filter IGH chains</td> <td>Filter stops in CDR3, OOF in CDR3</td> <td>Downsample by umi automatic</td> <td>590</td> <td>172729.0</td> <td>213</td> <td>30927.0</td> <td>Filter IGH chains</td> <td>590</td> <td>172729.0</td> <td>238</td> <td>30975.0</td> <td>Filter stops in CDR3, OOF in CDR3</td> <td>238</td> <td>30975.0</td> <td>213</td> <td>30927.0</td> <td>Downsample by umi automatic</td> <td>213</td> <td>30927.0</td> <td>213</td> <td>30927.0</td> <td></td> <td></td> <td></td> </tr> <tr> <td>IsotypeUsage</td> <td>mouse_3_Cd4Cre.clns</td> <td>Filter IGH chains</td> <td>Filter stops in CDR3, OOF in CDR3</td> <td>Downsample by umi automatic</td> <td>667</td> <td>195492.0</td> <td>240</td> <td>46633.0</td> <td>Filter IGH chains</td> <td>667</td> <td>195492.0</td> <td>273</td> <td>46724.0</td> <td>Filter stops in CDR3, OOF in CDR3</td> <td>273</td> <td>46724.0</td> <td>240</td> <td>46633.0</td> <td>Downsample by umi automatic</td> <td>240</td> <td>46633.0</td> <td>240</td> <td>46633.0</td> <td></td> <td></td> <td></td> </tr> <tr> <td>IsotypeUsage</td> <td>mouse_3.clns</td> <td>Filter IGH chains</td> <td>Filter stops in CDR3, OOF in CDR3</td> <td>Downsample by umi automatic</td> <td>1565</td> <td>347309.0</td> <td>530</td> <td>46633.0</td> <td>Filter IGH chains</td> <td>1565</td> <td>347309.0</td> <td>651</td> <td>65731.0</td> <td>Filter stops in CDR3, OOF in CDR3</td> <td>651</td> <td>65731.0</td> <td>596</td> <td>65563.0</td> <td>Downsample by umi automatic</td> <td>596</td> <td>65563.0</td> <td>530</td> <td>46633.0</td> <td></td> <td></td> <td></td> </tr> <tr> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> <td>...</td> </tr> </tbody> </table> <p>Columns explained:</p> <dl> <dt><code>characteristic</code></dt> <dd>metrics name (ex.<code>IsotypeUsage</code>)</dd> <dt><code>sample</code></dt> <dd>name of the <code>.clns</code> file (ex. <code>mouse_4_Cd4Cre.clns</code>)</dd> <dt><code>preprocessor</code></dt> <dd> <p>the name of the <em>overall</em> preprocessors chain applied to the dataset. <code>FilterIGHchains|FilterstopsinCDR3,OOFinCDR3|Downsampleautomatic</code> means the data has been subjected to three consecutive preprocessors:</p> <ul> <li>Only IGH chains were filtered</li> <li>Non-functional clonotypes containing stop-codons and OOF clones were filltered out and only functional were left</li> <li>Automatic Downsampling function has been applied to normalize sample sizes</li> </ul> </dd> <dt><code>nElementsBefore</code></dt> <dd>number of clonotypes before any preprocessing applied (that is in the initial dataset) (ex. <code>590</code> clones)</dd> <dt><code>sumWeightBefore</code></dt> <dd>total weight of all clonotypes before any preprocessing applied. The weight may be either number of reads, UMIs or cells, depending on the selected downsampling. In this case it represents the total number of reads that contribute to clonotypes (ex.<code>172729.0</code> reads)</dd> <dt><code>nElementsAfter</code></dt> <dd>number of clonotypes in the dataset after all preprocessors have been applied (ex.<code>213</code> clones)</dd> <dt><code>sumWeightAfter</code></dt> <dd>total weight of all clonotypes after all preprocessors have been applied (ex. <code>30927.0</code>). Note that all samples will have the same value for <code>sumWeightAfter</code> in this case, because downsampling was applied, thus all samples have been normalized to the same weight.</dd> <dt><code>preprocessor#i</code></dt> <dd>i-th part of the preprocessing chain (ex. <code>preprocessor#1</code>: <code>Filter IGH chains</code>)</dd> <dt><code>nElementsBefore#i</code></dt> <dd>number of clonotypes before i-th preprocessor has been applied (ex. <code>nElementsBefore#1</code>: <code>590</code>)</dd> <dt><code>sumWeightBefore#i</code></dt> <dd>total weight of all clonotypes before i-th preprocessor has been applied (ex. <code>sumWeightBefore#1</code>: <code>172729.0</code>)</dd> <dt><code>nElementsAfter#i</code></dt> <dd>number of clonotypes after i-th preprocessor has been applied (ex. <code>nElementsAfter#1</code>: <code>238</code>)</dd> <dt><code>sumWeightAfter#i</code></dt> <dd>total weight of all clonotypes after i-th preprocessor has been applied (ex. <code>sumWeightAfter#1</code>: <code>30975.0</code>)</dd> </dl> <p>. . .</p> <p>Various postanalysis tables contain information about each metric computed for each sample. For example, let's have a look inside <code>pa.i.diversity.IGK.tsv</code>. This table contains frequencies for each J segment present in the sample :</p> <table> <thead> <tr> <th>Observed diversity</th> <th>Shannon-Wiener diversity</th> <th>Normalized Shannon-Wiener index</th> <th>Inverse Simpson index</th> <th>Gini index</th> <th>Chao1 estimate</th> <th>Efron-Thisted estimate</th> <th>d50</th> <th></th> </tr> </thead> <tbody> <tr> <td>mouse_1_Cd4Cre.clns</td> <td>354.0</td> <td>353.99999999999994</td> <td>1.0</td> <td>354.0</td> <td>1.002824858757062</td> <td>62835.0</td> <td>6018.0</td> <td>6.0</td> </tr> <tr> <td>mouse_1.clns</td> <td>731.0</td> <td>730.9999999999997</td> <td>1.0</td> <td>731.0</td> <td>1.0013679890560876</td> <td>267546.0</td> <td>15351.0</td> <td>15.0</td> </tr> <tr> <td>mouse_2_Cd4Cre.clns</td> <td>393.0</td> <td>393.0</td> <td>1.0</td> <td>392.99999999999994</td> <td>1.0025445292620865</td> <td>77421.0</td> <td>8253.0</td> <td>7.0</td> </tr> <tr> <td>mouse_2.clns</td> <td>751.0</td> <td>750.9999999999998</td> <td>1.0</td> <td>751.0</td> <td>1.0013315579227697</td> <td>282376.0</td> <td>15771.0</td> <td>11.0</td> </tr> <tr> <td>mouse_3_Cd4Cre.clns</td> <td>304.0</td> <td>304.00000000000006</td> <td>1.0</td> <td>304.0</td> <td>1.0032894736842106</td> <td>46360.0</td> <td>2432.0</td> <td>10.0</td> </tr> <tr> <td>mouse_3.clns</td> <td>472.0</td> <td>471.99999999999983</td> <td>1.0</td> <td>472.0</td> <td>1.0021186440677967</td> <td>111628.0</td> <td>9912.0</td> <td>6.0</td> </tr> <tr> <td>mouse_4_Cd4Cre.clns</td> <td>257.0</td> <td>257.00000000000006</td> <td>1.0</td> <td>257.0</td> <td>1.0038910505836576</td> <td>33153.0</td> <td>1542.0</td> <td>8.0</td> </tr> <tr> <td>mouse_4.clns</td> <td>263.0</td> <td>263.00000000000006</td> <td>1.0</td> <td>263.0</td> <td>1.0038022813688212</td> <td>34716.0</td> <td>1578.0</td> <td>5.0</td> </tr> </tbody> </table> <h4 id="graphical-output">Graphical output</h4> <p>MiXCR allows to export graphical results in PDF, EPS, PNG and SVG formats using <code>exportPlots</code> command.</p> <p>For diversity metrics and CDR3 properties MiXCR allows to group data in different ways according to the submitted <code>metadata.tsv</code> and apply various statistical tests for group comparison.</p> <h5 id="diversity">Diversity</h5> <p>Let's reconstruct one of the figures from the paper. Let's say we want to look at the Normalized ShannonWiener diversity index. We will group samples by tissues and use separate facets for contol and experiment group. That can be easily done with a single command:</p> <div class="highlight"><pre><span></span><code>&gt; mixcr exportPlots diversity <span class="se">\</span>
    --metadata metadata.tsv <span class="se">\</span>
    --plot-type boxplot <span class="se">\</span>
    --metric normalizedShannonWienerIndex <span class="se">\ </span>
    --primary-group genotype <span class="se">\</span>
    postanalysis/individual.json.gz <span class="se">\</span>
    normalizedShannonWienerIndex.pdf
</code></pre></div> <figure> <p><a class="glightbox" data-desc-position="bottom" data-description="" data-height="auto" data-title="" data-width="100%" href="../generic-multiplex-bcr/normalizedShannonWienerIndex.IGH.svg"><img alt="normalizedShannonWienerIndex.IGH.svg" src="../generic-multiplex-bcr/normalizedShannonWienerIndex.IGH.svg"/></a></p> </figure> <p>Arguments explained:</p> <dl> <dt><code>--metadata</code></dt> <dd><code>metadata.tsv</code> is the name of metadata file.</dd> <dt><code>--plot-type</code></dt> <dd><code>boxplot</code> is the type of plot to generate. See <a href="../../reference/mixcr-exportPlots/"><code>mixcr exportPlots</code></a> for other options.</dd> <dt><code>--metric</code>:</dt> <dd><code>normalizedShannonWienerIndex</code> is the name of the metric to visualize. See <a href="../../reference/mixcr-exportPlots/#diversity-and-cdr3-metrics"><code>diversity metrics</code></a> for other options. If not specified multiple plots will be generated for each available diversity metric.</dd> <dt><code>--primary-group</code></dt> <dd><code>genotype</code> is name of the column with metadata values from <code>metadata.tsv</code> to group samples by.</dd> <dt><code>postanalysis/individual.json.gz</code></dt> <dd>the name of gzipped JSON file with all postanalysis results generated by <a href="../../reference/mixcr-postanalysis/#individual-postanalysis">`mixcr postanalysis individual</a></dd> <dt><code>normalizedShannonWienerIndex.pdf</code></dt> <dd>the name of output file. Also specifies the extension of the output file. One can use one of the following graphical out formats: <code>.pdf</code>, <code>.eps</code>, <code>.png</code> and <code>.svg</code>.</dd> </dl> <h5 id="v-usage">V usage</h5> <p>Now lets look at the J gene distribution among all samples.</p> <div class="highlight"><pre><span></span><code>&gt; mixcr exportPlots jUsage <span class="se">\</span>
    --metadata metadata.tsv <span class="se">\</span>
    --chains IGH <span class="se">\</span>
    --palette sequential <span class="se">\</span>
    --color-key tissue <span class="se">\</span>
    postanalysis/individual.json.gz <span class="se">\</span>
    jUsage.pdf  
</code></pre></div> <dl> <dt><code>--palette</code></dt> <dd><code>sequential</code>. Determines color palette to be used for the heatmap. See <a href="../../reference/mixcr-exportPlots/#gene-segment-usage">gene segment usage plots</a> for other options.</dd> <dt><code>--color-key</code></dt> <dd><code>tissue</code>. Metadata column name.</dd> </dl> <figure> <p><a class="glightbox" data-desc-position="bottom" data-description="" data-height="auto" data-title="" data-width="100%" href="../generic-multiplex-bcr/jUsage.IGH.svg"><img alt="jUsage.IGH.svg" src="../generic-multiplex-bcr/jUsage.IGH.svg"/></a></p> </figure> <h3 id="overlap-postanalysis">Overlap postanalysis</h3> <h4 id="two-samples-overlap">Two samples overlap</h4> <p>Since our samples were prepared in replicas, it is often usefully to check if clone frequencies correlate between replicas. Let's take two replicas of one biological sample and overlap two repertoires. We will use <a href="../../reference/mixcr-overlapScatterPlot/"><code>mixcr overlapScatterPlot</code> function</a>:</p> <div class="highlight"><pre><span></span><code>&gt; mixcr overlapScatterPlot <span class="se">\</span>
    --downsampling none <span class="se">\</span>
    --chains IGH <span class="se">\</span>
    results/M1_4T1_Blood_S2.clns results/M1_4T1_Blood_S6.clns <span class="se">\</span>
    M1_4T1_Blood.overlap.pdf
</code></pre></div> <figure> <p><a class="glightbox" data-desc-position="bottom" data-description="" data-height="auto" data-title="" data-width="100%" href="../generic-multiplex-bcr/M1_4T1_Blood.overlap.png"><img alt="M1_4T1_Blood.overlap.png" src="../generic-multiplex-bcr/M1_4T1_Blood.overlap.png"/></a></p> </figure> <h4 id="all-vs-all-overlap">All-vs-All overlap</h4> <p>MiXCR also allows performing an overall overlap analysis using <a href="../../reference/mixcr-postanalysis/#overlap-postanalysis"><code>mixcr postanalysis overlap</code></a>. But here, since there are a lot of samples we want to actually overlap groups of samples. Running the following command will perform pairwise overlap comparison between groups of samples with different <code>tissue</code> and <code>condition</code> values.</p> <div class="highlight"><pre><span></span><code>&gt; mixcr postanalysis overlap -f <span class="se">\</span>
   --factor-by tissue,condition <span class="se">\</span>
   --metadata metadata.tsv <span class="se">\</span>
   --default-downsampling count-reads-auto <span class="se">\</span>
   --default-weight-function <span class="nb">read</span> <span class="se">\</span>
   --only-productive <span class="se">\</span>
   --tables postanalysis/postanalysis.overlap.tsv <span class="se">\</span>
   --preproc-tables postanalysis/preproc.overlap.tsv <span class="se">\</span>
   results/*.clns <span class="se">\</span>
   postanalysisls /overlap.tissue_condition.json.gz
</code></pre></div> <p>This command will generate a set of files:</p> <div class="highlight"><pre><span></span><code><span class="c1">#Tsv files for every metric</span>
postanalysis.overlap.F1Index.IGH.tsv
postanalysis.overlap.F2Index.IGH.tsv
postanalysis.overlap.JaccardIndex.IGH.tsv
postanalysis.overlap.PearsonAll.IGH.tsv
postanalysis.overlap.Pearson.IGH.tsv
postanalysis.overlap.RelativeDiversity.IGH.tsv
postanalysis.overlap.SharedClonotypes.IGH.tsv

<span class="c1"># summary of applied preprocessors</span>
preproc.overlap.IGH.tsv

<span class="c1"># gzipped JSON with all results </span>
overlap.tissue_condition.json.gz
</code></pre></div> <p>The tabular output for example for F2 metric will look like:</p> <table> <thead> <tr> <th></th> <th>LN,control</th> <th>BM,4T1</th> <th>BM,control</th> <th>DLN,4T1</th> <th>Blood,4T1</th> <th>Blood,control</th> <th>Tumor,4T1</th> </tr> </thead> <tbody> <tr> <td><strong>LN,control</strong></td> <td>1.0</td> <td>0.013774364765374518</td> <td>0.04650437588957222</td> <td>0.014099278585204775</td> <td>0.031036354127937787</td> <td>0.027203139358090662</td> <td>0.010163687504766344</td> </tr> <tr> <td><strong>BM,4T1</strong></td> <td>0.013774364765374518</td> <td>1.0</td> <td>0.05529474567524548</td> <td>0.04405383350942532</td> <td>0.10645680248929848</td> <td>0.07417602888238424</td> <td>0.09466939004888049</td> </tr> <tr> <td><strong>BM,control</strong></td> <td>0.04650437588957222</td> <td>0.05529474567524548</td> <td>1.0</td> <td>0.03286381709464843</td> <td>0.233349361966603</td> <td>0.11870207376989239</td> <td>0.06314011150401469</td> </tr> <tr> <td><strong>DLN,4T1</strong></td> <td>0.014099278585204775</td> <td>0.04405383350942532</td> <td>0.03286381709464843</td> <td>1.0</td> <td>0.1768279650269904</td> <td>0.0188038306990065</td> <td>0.10982393111366022</td> </tr> <tr> <td><strong>Blood,4T1</strong></td> <td>0.031036354127937787</td> <td>0.10645680248929848</td> <td>0.233349361966603</td> <td>0.1768279650269904</td> <td>1.0</td> <td>0.0634481121480744</td> <td>0.13415831269714423</td> </tr> <tr> <td><strong>Blood,control</strong></td> <td>0.027203139358090662</td> <td>0.07417602888238424</td> <td>0.11870207376989239</td> <td>0.0188038306990065</td> <td>0.0634481121480744</td> <td>1.0</td> <td>0.03867881035646659</td> </tr> <tr> <td><strong>Tumor,4T1</strong></td> <td>0.010163687504766344</td> <td>0.09466939004888049</td> <td>0.06314011150401469</td> <td>0.10982393111366022</td> <td>0.13415831269714423</td> <td>0.03867881035646659</td> <td>1.0</td> </tr> </tbody> </table> <p>Every overlap metric is also possible to present in a graphical format:</p> <div class="highlight"><pre><span></span><code>&gt; mixcr exportPlots overlap <span class="se">\</span>
    --chains IGH <span class="se">\</span>
    --palette density <span class="se">\</span>
    --metric f2Index <span class="se">\</span>
    postanalysis/overlap.tissue_condition.json.gz <span class="se">\</span>
    overlap.time_marker.pdf
</code></pre></div> <p>For list of available metrics see <a href="../../reference/mixcr-exportPlots/#overlap"><code>mixcr exportPlots overlap</code></a></p> <figure> <p><a class="glightbox" data-desc-position="bottom" data-description="" data-height="auto" data-title="" data-width="100%" href="../generic-multiplex-bcr/overlap.tissue_condition.IGH.svg"><img alt="overlap.tissue_condition.IGH.svg" src="../generic-multiplex-bcr/overlap.tissue_condition.IGH.svg"/></a></p> </figure> <p>For further details see <a href="../../reference/mixcr-exportPlots/">overlap postanalysis reference</a>.</p> </article> <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script> </div> </div> <a class="md-top md-icon" data-md-component="top" hidden="" href="#"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg> Back to top </a> </main> <footer class="md-footer"> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class="md-copyright"> <div class="md-copyright__highlight"> Copyright © MiLaboratories Inc </div> </div> </div> </div> </footer> </div> <div class="md-dialog" data-md-component="dialog"> <div class="md-dialog__inner md-typeset"></div> </div> <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.instant", "navigation.tabs", "navigation.top", "navigation.sections", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "content.tabs.link"], "search": "../../../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search by name or category", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "version": {"default": "latest", "provider": "mike"}}</script> <script src="../../../assets/javascripts/bundle.6c7ad80a.min.js"></script> <script src="../../../javascripts/svg.js"></script> <script src="../../../javascripts/onload.js"></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script> <script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom"});})</script></body> </html>