<!DOCTYPE html>
<html class="no-js" lang="en"> <head><meta charset="utf-8"/><meta content="width=device-width,initial-scale=1" name="viewport"/><link href="https://docs.milaboratories.com/mixcr/guides/abhelix-bcr/" rel="canonical"/><link href="../../../assets/favicon.png" rel="icon"/><meta content="mkdocs-1.3.1, mkdocs-material-8.3.9" name="generator"/><title>AbHelix BCR - Docs @ MiLaboratories</title><link href="../../../assets/stylesheets/main.1d29e8d0.min.css" rel="stylesheet"/><link href="../../../assets/stylesheets/palette.cbb835fc.min.css" rel="stylesheet"/><link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/><link href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,300i,400,400i,700,700i%7CIBM+Plex+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/><style>:root{--md-text-font:"IBM Plex Sans";--md-code-font:"IBM Plex Mono"}</style><link href="../../../css/timeago.css" rel="stylesheet"/><link href="../../../css/extra.css" rel="stylesheet"/><script>__md_scope=new URL("../../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-1KLG43HZP1"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){this.value&&gtag("event","search",{search_term:this.value})}),"undefined"!=typeof location$&&location$.subscribe(function(e){gtag("config","G-1KLG43HZP1",{page_path:e.pathname})})})</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-1KLG43HZP1"></script><link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>html.glightbox-open { overflow: initial; height: 100%; }</style><script src="../../../assets/javascripts/glightbox.min.js"></script></head> <body data-md-color-accent="" data-md-color-primary="" data-md-color-scheme="default" dir="ltr"> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/> <input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/> <label class="md-overlay" for="__drawer"></label> <div data-md-component="skip"> <a class="md-skip" href="#abhelix"> Skip to content </a> </div> <div data-md-component="announce"> </div> <header class="md-header" data-md-component="header"> <nav aria-label="Header" class="md-header__inner md-grid"> <a aria-label="Docs @ MiLaboratories" class="md-header__button md-logo" data-md-component="logo" href="../../.." title="Docs @ MiLaboratories"> <img alt="logo" src="../../../assets/logo.svg"/> </a> <label class="md-header__button md-icon" for="__drawer"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg> </label> <div class="md-header__title" data-md-component="header-title"> <div class="md-header__ellipsis"> <div class="md-header__topic"> <span class="md-ellipsis"> Docs @ MiLaboratories </span> </div> <div class="md-header__topic" data-md-component="header-topic"> <span class="md-ellipsis"> AbHelix BCR </span> </div> </div> </div> <form class="md-header__option" data-md-component="palette"> <input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/> <label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to dark mode"> <svg fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <path clip-rule="evenodd" d="M18.3578 4.32112C18.2104 4.39482 18.2104 4.60518 18.3578 4.67889L18.9404 4.97019C18.9791 4.98954 19.0105 5.02092 19.0298 5.05963L19.3211 5.64223C19.3948 5.78964 19.6052 5.78964 19.6789 5.64223L19.9702 5.05963C19.9895 5.02092 20.0209 4.98954 20.0596 4.97019L20.6422 4.67889C20.7896 4.60518 20.7896 4.39482 20.6422 4.32112L20.0596 4.02982C20.0209 4.01046 19.9895 3.97908 19.9702 3.94037L19.6789 3.35777C19.6052 3.21036 19.3948 3.21036 19.3211 3.35777L19.0298 3.94037C19.0105 3.97908 18.9791 4.01046 18.9404 4.02982L18.3578 4.32112ZM18.2083 16.2099C17.5036 16.3991 16.7631 16.5 16 16.5C11.3056 16.5 7.5 12.6944 7.5 8C7.5 7.23686 7.60088 6.4964 7.79006 5.79169C5.80401 7.14138 4.5 9.41947 4.5 12C4.5 16.1421 7.85786 19.5 12 19.5C14.5805 19.5 16.8586 18.196 18.2083 16.2099ZM19.1573 14.2492C18.2087 14.7294 17.136 15 16 15C12.134 15 9 11.866 9 8C9 6.86405 9.27058 5.79129 9.75081 4.84267L9.77821 4.78912C9.83867 4.6722 9.90234 4.55719 9.96908 4.44423C10.0299 4.34128 10.0933 4.24002 10.1592 4.14055C10.4192 3.74773 10.1191 3.18416 9.66405 3.30612C9.48653 3.3537 9.31118 3.40657 9.13817 3.46455C9.01891 3.50452 8.90077 3.54692 8.7838 3.59169C8.76089 3.60045 8.73803 3.60931 8.71521 3.61826C5.3692 4.93057 3 8.18861 3 12C3 16.9706 7.02944 21 12 21C15.8114 21 19.0694 18.6308 20.3817 15.2848C20.3907 15.262 20.3995 15.2391 20.4083 15.2162C20.4531 15.0992 20.4955 14.9811 20.5354 14.8618C20.5934 14.6888 20.6463 14.5135 20.6939 14.3359C20.8158 13.8809 20.2523 13.5808 19.8595 13.8408C19.76 13.9067 19.6587 13.9701 19.5558 14.0309C19.4428 14.0977 19.3278 14.1613 19.2109 14.2218L19.1573 14.2492ZM16.4602 7.40052L15.7321 5.5803C15.6483 5.37076 15.3517 5.37076 15.2679 5.5803L14.5398 7.40052C14.5144 7.46405 14.464 7.51438 14.4005 7.53979L12.5803 8.26788C12.3708 8.3517 12.3708 8.64831 12.5803 8.73212L14.4005 9.46021C14.464 9.48562 14.5144 9.53596 14.5398 9.59948L15.2679 11.4197C15.3517 11.6292 15.6483 11.6292 15.7321 11.4197L16.4602 9.59948C16.4856 9.53596 16.536 9.48562 16.5995 9.46021L18.4197 8.73212C18.6292 8.64831 18.6292 8.3517 18.4197 8.26788L16.5995 7.53979C16.536 7.51438 16.4856 7.46405 16.4602 7.40052Z" fill="white" fill-rule="evenodd"></path> </svg> </label> <input aria-label="Switch to light mode" class="md-option" data-md-color-accent="" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="" data-md-color-scheme="slate" id="__palette_2" name="__palette" type="radio"/> <label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to light mode"> <svg fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <path clip-rule="evenodd" d="M12.75 3V0H11.25V3H12.75ZM12.75 24V21H11.25V24H12.75ZM3 12.75H0V11.25H3V12.75ZM21 12.75H24V11.25H21V12.75ZM12 16.5C14.4853 16.5 16.5 14.4853 16.5 12C16.5 9.51472 14.4853 7.5 12 7.5C9.51472 7.5 7.5 9.51472 7.5 12C7.5 14.4853 9.51472 16.5 12 16.5ZM12 18C15.3137 18 18 15.3137 18 12C18 8.68629 15.3137 6 12 6C8.68629 6 6 8.68629 6 12C6 15.3137 8.68629 18 12 18ZM21.0156 4.04506L18.8943 6.16638L17.8336 5.10572L19.955 2.9844L21.0156 4.04506ZM5.10571 17.8336L6.16637 18.8943L4.04505 21.0156L2.98439 19.955L5.10571 17.8336ZM2.98439 4.04506L5.10571 6.16638L6.16637 5.10572L4.04505 2.9844L2.98439 4.04506ZM21.0156 19.955L19.955 21.0156L17.8336 18.8943L18.8943 17.8336L21.0156 19.955Z" fill="white" fill-rule="evenodd"></path> </svg> </label> </form> <label class="md-header__button md-icon" for="__search"> <svg class="magnify" fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <circle cx="11" cy="11" r="7.25" stroke="white" stroke-width="1.5"></circle> <path d="M16 16L21.5 21.5" stroke="white" stroke-width="1.5"></path> </svg> </label> <div class="md-search" data-md-component="search" role="dialog"> <label class="md-search__overlay" for="__search"></label> <div class="md-search__inner" role="search"> <form class="md-search__form" name="search"> <input aria-label="Search by name or category" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search by name or category" required="" spellcheck="false" type="text"/> <label class="md-search__icon md-icon" for="__search"> <svg class="magnify" fill="none" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"> <circle cx="11" cy="11" r="7.25" stroke="white" stroke-width="1.5"></circle> <path d="M16 16L21.5 21.5" stroke="white" stroke-width="1.5"></path> </svg> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg> </label> <nav aria-label="Search" class="md-search__options"> <a aria-label="Share" class="md-search__icon md-icon" data-clipboard="" data-clipboard-text="" data-md-component="search-share" href="javascript:void(0)" tabindex="-1"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7 0-.24-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91 1.61 0 2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08Z"></path></svg> </a> <button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" type="reset"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg> </button> </nav> <div class="md-search__suggest" data-md-component="search-suggest"></div> </form> <div class="md-search__output"> <div class="md-search__scrollwrap" data-md-scrollfix=""> <div class="md-search-result" data-md-component="search-result"> <div class="md-search-result__meta"> Initializing search </div> <ol class="md-search-result__list"></ol> </div> </div> </div> </div> </div> <div class="md-header__source"> <a class="md-source" data-md-component="source" href="https://github.com/milaboratory/docs" title="Go to repository"> <div class="md-source__icon md-icon"> <svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg> </div> <div class="md-source__repository"> milaboratory/docs </div> </a> </div> </nav> </header> <div class="md-container" data-md-component="container"> <nav aria-label="Tabs" class="md-tabs" data-md-component="tabs"> <div class="md-tabs__inner md-grid"> <ul class="md-tabs__list"> <li class="md-tabs__item"> <a class="md-tabs__link" href="../../.."> Home </a> </li> <li class="md-tabs__item"> <a class="md-tabs__link" href="../../getting-started/installation/"> Getting Started </a> </li> <li class="md-tabs__item"> <a class="md-tabs__link md-tabs__link--active" href="../milaboratories-human-tcr-rna-multi/"> Guides </a> </li> <li class="md-tabs__item"> <a class="md-tabs__link" href="../../reference/mixcr-analyze/"> Reference </a> </li> </ul> </div> </nav> <main class="md-main" data-md-component="main"> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation"> <div class="md-sidebar__scrollwrap"> <div class="md-sidebar__inner"> <!-- Determine class according to configuration --> <!-- Ugly hack --> <!-- Main navigation --> <nav aria-label="Navigation" class="md-nav md-nav--primary md-nav--lifted has-section" data-md-level="0" data-test="1"> <!-- Site title --> <label class="md-nav__title" for="__drawer"> <a aria-label="Docs @ MiLaboratories" class="md-nav__button md-logo" data-md-component="logo" href="../../.." title="Docs @ MiLaboratories"> <img alt="logo" src="../../../assets/logo.svg"/> </a> Docs @ MiLaboratories </label> <!-- Repository information --> <div class="md-nav__source"> <a class="md-source" data-md-component="source" href="https://github.com/milaboratory/docs" title="Go to repository"> <div class="md-source__icon md-icon"> <svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg> </div> <div class="md-source__repository"> milaboratory/docs </div> </a> </div> <!-- Render item list --> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../.."> Home </a> </li> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" id="__nav_2" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="1" for="__nav_2"> Getting Started <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Getting Started" class="md-nav" data-md-level="1"> <label class="md-nav__title" for="__nav_2"> <span class="md-nav__icon md-icon"></span> Getting Started </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/installation/"> Installation </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/docker/"> Using with Docker </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/milm/"> Obtaining and using license key </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/overview/"> MiXCR Overview </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../getting-started/license/"> License </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input checked="" class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" id="__nav_3" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="1" for="__nav_3"> Guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Guides" class="md-nav" data-md-level="1"> <label class="md-nav__title" for="__nav_3"> <span class="md-nav__icon md-icon"></span> Guides </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input checked="" class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_1" id="__nav_3_1" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_3_1"> Kits &amp; Protocols <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Kits &amp; Protocols" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_3_1"> <span class="md-nav__icon md-icon"></span> Kits &amp; Protocols </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../milaboratories-human-tcr-rna-multi/"> MiLaboratories Human TCR RNA Multiplex kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../biomed2-bcr/"> BioMed2 </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item md-nav__item--active"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="toc" id="__toc" type="checkbox"/> <!-- Hack: see partials/toc.html for more information --> <!-- Render table of contents, if not empty --> <label class="md-nav__link md-nav__link--active" for="__toc"> AbHelix BCR <span class="md-nav__icon md-icon"></span> </label> <a class="md-nav__link md-nav__link--active" href="./"> AbHelix BCR </a> <!-- Show table of contents --> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav aria-label="Table of contents" class="md-nav md-nav--secondary hello"> <label class="md-nav__title" for="__toc"> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix=""> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#data-libraries"> Data libraries </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#upstream-analysis"> Upstream analysis </a> <!-- Table of contents list --> <nav aria-label="Upstream analysis" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#under-the-hood-pipeline"> Under the hood pipeline: </a> <!-- Table of contents list --> <nav aria-label="Under the hood pipeline:" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#align"> align </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#assemble"> assemble </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#export"> export </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#quality-control"> Quality control </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#advanced-settings"> Advanced settings </a> <!-- Table of contents list --> </li> </ul> </nav> </li> <!-- Main navigation item --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../takara-mmu-tcr/"> SMARTer Mouse TCR a/b Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../takara-hsa-bcr/"> SMARTer Human BCR IgG IgM H/K/L Profiling Kit </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../nebnext-bcr/"> NEBNext Immune Sequencing </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../qiaseq-tcr/"> QIAseq Immune Repertoire TCR </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../rnaseq/"> RNA-Seq data </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_2" id="__nav_3_2" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_3_2"> Analysis cases <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Analysis cases" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_3_2"> <span class="md-nav__icon md-icon"></span> Analysis cases </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../generic-umi-tcr/"> Targeted TCR UMI libraries </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../generic-multiplex-bcr/"> Targeted BCR Multiplex libraries </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_3" id="__nav_3_3" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_3_3"> Special guides <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Special guides" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_3_3"> <span class="md-nav__icon md-icon"></span> Special guides </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../create-custom-library/"> Assembling custom reference library </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../external-libraries/"> Using external libraries </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> <!-- Wrap everything with a macro to reduce file roundtrips (see #2213) --> <!-- Render current and nested navigation items --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" id="__nav_4" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="1" for="__nav_4"> Reference <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Reference" class="md-nav" data-md-level="1"> <label class="md-nav__title" for="__nav_4"> <span class="md-nav__icon md-icon"></span> Reference </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_1" id="__nav_4_1" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_1"> MiXCR <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="MiXCR" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_1"> <span class="md-nav__icon md-icon"></span> MiXCR </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-analyze/"> analyze </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-align/"> align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-correctAndSortTags/"> correctAndSortTags </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-assemblePartial/"> assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-extend/"> extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-assemble/"> assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-assembleContigs/"> assembleContigs </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-export/"> export </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportPretty/"> exportPretty </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportAirr/"> exportAirr </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportReadsForClones/"> exportReadsForClones </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportReports/"> exportReports </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportQc/"> exportQc </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-postanalysis/"> postanalysis </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportPlots/"> exportPlots </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportTables/"> exportTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportPreprocTables/"> exportPreprocTables </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-exportClonesOverlap/"> exportClonesOverlap </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mixcr-overlapScatterPlot/"> overlapScatterPlot </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" id="__nav_4_2" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_2"> Reports <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Reports" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_2"> <span class="md-nav__icon md-icon"></span> Reports </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-align/"> mixcr align </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-correctAndSortTags/"> mixcr correctAndSortTags </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-assemblePartial/"> mixcr assemblePartial </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-extend/"> mixcr extend </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-assemble/"> mixcr assemble </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/report-assembleContigs/"> mixcr assembleContigs </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_3" id="__nav_4_3" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_3"> RepSeq.io <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="RepSeq.io" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_3"> <span class="md-nav__icon md-icon"></span> RepSeq.io </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-fasta/"> fasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-compile/"> compile </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-debug/"> debug </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-filter/"> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-format/"> format </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-fromFasta/"> fromFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-fromPaddedFasta/"> fromPaddedFasta </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-inferPoints/"> inferPoints </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-merge/"> merge </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-stat/"> stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-tsv/"> tsv </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_4" id="__nav_4_4" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_4"> MiTool <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="MiTool" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_4"> <span class="md-nav__icon md-icon"></span> MiTool </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-parse/"> parse </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-filter/"> filter </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-tag-stat/"> tag-stat </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-export-fastq/"> export-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-correct/"> correct </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-consensus/"> consensus </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-process-fastq/"> process-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-format-fastq/"> format-fastq </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/mitool-compress-sequence-set/"> compress-sequence-set </a> </li> </ul> </nav> </li> <!-- Currently active page --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <!-- Determine whether to render item as a section --> <!-- Render item with nested items --> <li class="md-nav__item md-nav__item--section md-nav__item--nested red"> <!-- Active checkbox expands items contained within nested section --> <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_5" id="__nav_4_5" type="checkbox"/> <!-- Determine all nested items that are index pages --> <!-- Render toggle to expand nested items --> <label class="md-nav__link" data-level="2" for="__nav_4_5"> Specs <span class="md-nav__icon md-icon"></span> </label> <!-- Render link to index page + toggle --> <!-- Render nested navigation --> <nav aria-label="Specs" class="md-nav" data-md-level="2"> <label class="md-nav__title" for="__nav_4_5"> <span class="md-nav__icon md-icon"></span> Specs </label> <ul class="md-nav__list" data-md-scrollfix=""> <!-- Render nested item list --> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/ref-tag-pattern/"> Tag Pattern </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/ref-gene-features/"> Gene features and reference points </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/ref-translation-rules/"> Translation rules </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/ref-mutations-encoding/"> Mutations encoding </a> </li> <!-- Determine class according to state --> <!-- Main navigation item with nested items --> <li class="md-nav__item green"> <a class="md-nav__link" href="../../reference/repseqio-json-format/"> RepSeq.io JSON format </a> </li> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </li> <!-- Currently active page --> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc"> <div class="md-sidebar__scrollwrap"> <div class="md-sidebar__inner"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Determine title --> <!-- Table of contents --> <!--
  Hack: check whether the content contains a h1 headline. If it does, the
  top-level anchor must be skipped, since it would be redundant to the link
  to the current page that is located just above the anchor. Therefore we
  directly continue with the children of the anchor.
--> <!-- Table of contents title and list --> <nav aria-label="Table of contents" class="md-nav md-nav--secondary hello"> <label class="md-nav__title" for="__toc"> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix=""> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#data-libraries"> Data libraries </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#upstream-analysis"> Upstream analysis </a> <!-- Table of contents list --> <nav aria-label="Upstream analysis" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#under-the-hood-pipeline"> Under the hood pipeline: </a> <!-- Table of contents list --> <nav aria-label="Under the hood pipeline:" class="md-nav"> <ul class="md-nav__list"> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#align"> align </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#assemble"> assemble </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#export"> export </a> <!-- Table of contents list --> </li> </ul> </nav> </li> </ul> </nav> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#quality-control"> Quality control </a> <!-- Table of contents list --> </li> <!--
  Copyright (c) 2016-2022 Martin Donath <martin.donath@squidfunk.com>

  Permission is hereby granted, free of charge, to any person obtaining a copy
  of this software and associated documentation files (the "Software"), to
  deal in the Software without restriction, including without limitation the
  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
  sell copies of the Software, and to permit persons to whom the Software is
  furnished to do so, subject to the following conditions:

  The above copyright notice and this permission notice shall be included in
  all copies or substantial portions of the Software.

  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
  IN THE SOFTWARE.
--> <!-- Table of contents item --> <li class="md-nav__item iiitem"> <a class="md-nav__link" href="#advanced-settings"> Advanced settings </a> <!-- Table of contents list --> </li> </ul> </nav> </div> </div> </div> <div class="md-content" data-md-component="content"> <article class="md-content__inner md-typeset"> <a class="md-content__button md-icon" href="https://github.com/milaboratory/docs/edit/main/docs/mixcr/guides/abhelix-bcr.md" title="Edit this page"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"></path></svg> </a> <h1 id="abhelix">ABHelix</h1> <p>Here we will discuss how to process BCR cDNA libraries obtained with ABHelix kit.</p> <h2 id="data-libraries">Data libraries</h2> <p>This tutorial uses the data from the following publication: <em>High frequency of shared clonotypes in human B cell receptor repertoires.</em> Soto C et al., , Nature, 2019 Feb;566(7744):398-402 <a href="https://doi.org/10.1038/s41586-019-0934-8">doi: 10.1038/s41586-019-0934-8</a></p> <p>Peripheral blood samples were collected from three donors : HIP1 (female, 47 y.o.), HIP2 (male, 22 y.o.) and HIP3(male, 29 y.o.). Samples were collected in multiple replicas. Total RNA was extracted from PBMCs and multiple reactions of reverse transcription were combined and purified using magnetic beads. The purified RT products were divided evenly for the first round of PCR amplification specific to human IgG, IgL IgK, IgM, or IgA. The 5' multiplex PCR primers were designed within the leader sequences of each productive V gene and the 3' primers within the constant regions, but in close approximation to the J gene/constant region junctions. Second PCR was performed using leader and V-C junction primers. Subsequent amplicon libraries were sequenced using next-generation sequencing. Paired end sequencing was performed on Illumina HiSeq 2500, R1 and R2 are 250bp long.</p> <p>The data for this tutorial can be downloaded using the script bellow.</p> <div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/env bash</span>
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365277/SRR8365277_1.fastq.gz -o SRR8365277_HIP1_female_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365278/SRR8365278_1.fastq.gz -o SRR8365278_HIP1_female_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365279/SRR8365279_1.fastq.gz -o SRR8365279_HIP1_female_IgG3_R1.fastq.gz
</code></pre></div> <details class="note"> <summary>See full script</summary> <div class="highlight"><pre><span></span><code>curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365277/SRR8365277_1.fastq.gz -o SRR8365277_HIP1_female_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365278/SRR8365278_1.fastq.gz -o SRR8365278_HIP1_female_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365279/SRR8365279_1.fastq.gz -o SRR8365279_HIP1_female_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365280/SRR8365280_1.fastq.gz -o SRR8365280_HIP1_female_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365281/SRR8365281_1.fastq.gz -o SRR8365281_HIP1_female_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365282/SRR8365282_1.fastq.gz -o SRR8365282_HIP1_female_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365283/SRR8365283_1.fastq.gz -o SRR8365283_HIP1_female_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365284/SRR8365284_1.fastq.gz -o SRR8365284_HIP1_female_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365285/SRR8365285_1.fastq.gz -o SRR8365285_HIP2_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365286/SRR8365286_1.fastq.gz -o SRR8365286_HIP2_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365287/SRR8365287_1.fastq.gz -o SRR8365287_HIP2_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365288/SRR8365288_1.fastq.gz -o SRR8365288_HIP2_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365289/SRR8365289_1.fastq.gz -o SRR8365289_HIP2_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365290/SRR8365290_1.fastq.gz -o SRR8365290_HIP2_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365291/SRR8365291_1.fastq.gz -o SRR8365291_HIP2_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365292/SRR8365292_1.fastq.gz -o SRR8365292_HIP2_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365293/SRR8365293_1.fastq.gz -o SRR8365293_HIP3_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365294/SRR8365294_1.fastq.gz -o SRR8365294_HIP3_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365295/SRR8365295_1.fastq.gz -o SRR8365295_HIP3_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365296/SRR8365296_1.fastq.gz -o SRR8365296_HIP3_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365297/SRR8365297_1.fastq.gz -o SRR8365297_HIP2_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365298/SRR8365298_1.fastq.gz -o SRR8365298_HIP2_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365299/SRR8365299_1.fastq.gz -o SRR8365299_HIP3_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365300/SRR8365300_1.fastq.gz -o SRR8365300_HIP3_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365311/SRR8365311_1.fastq.gz -o SRR8365311_HIP1_female_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365312/SRR8365312_1.fastq.gz -o SRR8365312_HIP1_female_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365313/SRR8365313_1.fastq.gz -o SRR8365313_HIP1_female_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365314/SRR8365314_1.fastq.gz -o SRR8365314_HIP1_female_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365315/SRR8365315_1.fastq.gz -o SRR8365315_HIP1_female_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365316/SRR8365316_1.fastq.gz -o SRR8365316_HIP1_female_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365317/SRR8365317_1.fastq.gz -o SRR8365317_HIP1_female_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365318/SRR8365318_1.fastq.gz -o SRR8365318_HIP1_female_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365319/SRR8365319_1.fastq.gz -o SRR8365319_HIP1_female_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365320/SRR8365320_1.fastq.gz -o SRR8365320_HIP1_female_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365323/SRR8365323_1.fastq.gz -o SRR8365323_HIP3_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365332/SRR8365332_1.fastq.gz -o SRR8365332_HIP2_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365333/SRR8365333_1.fastq.gz -o SRR8365333_HIP2_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365334/SRR8365334_1.fastq.gz -o SRR8365334_HIP2_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365335/SRR8365335_1.fastq.gz -o SRR8365335_HIP2_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365336/SRR8365336_1.fastq.gz -o SRR8365336_HIP2_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365337/SRR8365337_1.fastq.gz -o SRR8365337_HIP2_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365338/SRR8365338_1.fastq.gz -o SRR8365338_HIP2_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365339/SRR8365339_1.fastq.gz -o SRR8365339_HIP2_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365340/SRR8365340_1.fastq.gz -o SRR8365340_HIP3_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365341/SRR8365341_1.fastq.gz -o SRR8365341_HIP3_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365342/SRR8365342_1.fastq.gz -o SRR8365342_HIP2_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365343/SRR8365343_1.fastq.gz -o SRR8365343_HIP2_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365344/SRR8365344_1.fastq.gz -o SRR8365344_HIP3_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365345/SRR8365345_1.fastq.gz -o SRR8365345_HIP3_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365346/SRR8365346_1.fastq.gz -o SRR8365346_HIP3_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365347/SRR8365347_1.fastq.gz -o SRR8365347_HIP3_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365349/SRR8365349_1.fastq.gz -o SRR8365349_HIP2_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365350/SRR8365350_1.fastq.gz -o SRR8365350_HIP2_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365351/SRR8365351_1.fastq.gz -o SRR8365351_HIP2_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365352/SRR8365352_1.fastq.gz -o SRR8365352_HIP2_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365353/SRR8365353_1.fastq.gz -o SRR8365353_HIP2_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365354/SRR8365354_1.fastq.gz -o SRR8365354_HIP2_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365355/SRR8365355_1.fastq.gz -o SRR8365355_HIP2_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365356/SRR8365356_1.fastq.gz -o SRR8365356_HIP2_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365357/SRR8365357_1.fastq.gz -o SRR8365357_HIP2_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365358/SRR8365358_1.fastq.gz -o SRR8365358_HIP2_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365359/SRR8365359_1.fastq.gz -o SRR8365359_HIP3_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365360/SRR8365360_1.fastq.gz -o SRR8365360_HIP3_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365361/SRR8365361_1.fastq.gz -o SRR8365361_HIP3_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365362/SRR8365362_1.fastq.gz -o SRR8365362_HIP3_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365363/SRR8365363_1.fastq.gz -o SRR8365363_HIP3_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365364/SRR8365364_1.fastq.gz -o SRR8365364_HIP3_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365365/SRR8365365_1.fastq.gz -o SRR8365365_HIP3_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365366/SRR8365366_1.fastq.gz -o SRR8365366_HIP3_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365367/SRR8365367_1.fastq.gz -o SRR8365367_HIP3_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365368/SRR8365368_1.fastq.gz -o SRR8365368_HIP3_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365369/SRR8365369_1.fastq.gz -o SRR8365369_HIP3_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365370/SRR8365370_1.fastq.gz -o SRR8365370_HIP2_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365371/SRR8365371_1.fastq.gz -o SRR8365371_HIP2_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365372/SRR8365372_1.fastq.gz -o SRR8365372_HIP3_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365373/SRR8365373_1.fastq.gz -o SRR8365373_HIP2_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365374/SRR8365374_1.fastq.gz -o SRR8365374_HIP2_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365375/SRR8365375_1.fastq.gz -o SRR8365375_HIP2_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365376/SRR8365376_1.fastq.gz -o SRR8365376_HIP2_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365377/SRR8365377_1.fastq.gz -o SRR8365377_HIP2_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365378/SRR8365378_1.fastq.gz -o SRR8365378_HIP2_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365379/SRR8365379_1.fastq.gz -o SRR8365379_HIP2_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365380/SRR8365380_1.fastq.gz -o SRR8365380_HIP2_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365381/SRR8365381_1.fastq.gz -o SRR8365381_HIP3_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365382/SRR8365382_1.fastq.gz -o SRR8365382_HIP3_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365383/SRR8365383_1.fastq.gz -o SRR8365383_HIP3_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365384/SRR8365384_1.fastq.gz -o SRR8365384_HIP3_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365385/SRR8365385_1.fastq.gz -o SRR8365385_HIP3_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365386/SRR8365386_1.fastq.gz -o SRR8365386_HIP3_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365387/SRR8365387_1.fastq.gz -o SRR8365387_HIP3_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365388/SRR8365388_1.fastq.gz -o SRR8365388_HIP3_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365389/SRR8365389_1.fastq.gz -o SRR8365389_HIP3_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365390/SRR8365390_1.fastq.gz -o SRR8365390_HIP3_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365391/SRR8365391_1.fastq.gz -o SRR8365391_HIP1_female_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365392/SRR8365392_1.fastq.gz -o SRR8365392_HIP1_female_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365393/SRR8365393_1.fastq.gz -o SRR8365393_HIP1_female_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365394/SRR8365394_1.fastq.gz -o SRR8365394_HIP1_female_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365395/SRR8365395_1.fastq.gz -o SRR8365395_HIP1_female_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365396/SRR8365396_1.fastq.gz -o SRR8365396_HIP1_female_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365397/SRR8365397_1.fastq.gz -o SRR8365397_HIP1_female_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365398/SRR8365398_1.fastq.gz -o SRR8365398_HIP1_female_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365399/SRR8365399_1.fastq.gz -o SRR8365399_HIP1_female_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365400/SRR8365400_1.fastq.gz -o SRR8365400_HIP1_female_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365401/SRR8365401_1.fastq.gz -o SRR8365401_HIP3_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365402/SRR8365402_1.fastq.gz -o SRR8365402_HIP3_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365413/SRR8365413_1.fastq.gz -o SRR8365413_HIP3_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365414/SRR8365414_1.fastq.gz -o SRR8365414_HIP3_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365415/SRR8365415_1.fastq.gz -o SRR8365415_HIP3_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365416/SRR8365416_1.fastq.gz -o SRR8365416_HIP3_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365417/SRR8365417_1.fastq.gz -o SRR8365417_HIP3_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365425/SRR8365425_1.fastq.gz -o SRR8365425_HIP3_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365426/SRR8365426_1.fastq.gz -o SRR8365426_HIP3_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365427/SRR8365427_1.fastq.gz -o SRR8365427_HIP3_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365428/SRR8365428_1.fastq.gz -o SRR8365428_HIP1_female_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365429/SRR8365429_1.fastq.gz -o SRR8365429_HIP1_female_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365430/SRR8365430_1.fastq.gz -o SRR8365430_HIP1_female_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365431/SRR8365431_1.fastq.gz -o SRR8365431_HIP1_female_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365432/SRR8365432_1.fastq.gz -o SRR8365432_HIP1_female_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365433/SRR8365433_1.fastq.gz -o SRR8365433_HIP1_female_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365442/SRR8365442_1.fastq.gz -o SRR8365442_HIP3_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365445/SRR8365445_1.fastq.gz -o SRR8365445_HIP3_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365451/SRR8365451_1.fastq.gz -o SRR8365451_HIP2_male_IgG1_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365452/SRR8365452_1.fastq.gz -o SRR8365452_HIP2_male_IgG2_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365453/SRR8365453_1.fastq.gz -o SRR8365453_HIP2_male_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365454/SRR8365454_1.fastq.gz -o SRR8365454_HIP2_male_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365455/SRR8365455_1.fastq.gz -o SRR8365455_HIP2_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365466/SRR8365466_1.fastq.gz -o SRR8365466_HIP3_male_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365472/SRR8365472_1.fastq.gz -o SRR8365472_HIP2_male_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365476/SRR8365476_1.fastq.gz -o SRR8365476_HIP2_male_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365477/SRR8365477_1.fastq.gz -o SRR8365477_HIP2_male_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365478/SRR8365478_1.fastq.gz -o SRR8365478_HIP1_female_IgL_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365479/SRR8365479_1.fastq.gz -o SRR8365479_HIP1_female_IgK_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365480/SRR8365480_1.fastq.gz -o SRR8365480_HIP1_female_IgA_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365481/SRR8365481_1.fastq.gz -o SRR8365481_HIP1_female_IgM_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365486/SRR8365486_1.fastq.gz -o SRR8365486_HIP1_female_IgG4_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365487/SRR8365487_1.fastq.gz -o SRR8365487_HIP1_female_IgG3_R1.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365277/SRR8365277_2.fastq.gz -o SRR8365277_HIP1_female_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365278/SRR8365278_2.fastq.gz -o SRR8365278_HIP1_female_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365279/SRR8365279_2.fastq.gz -o SRR8365279_HIP1_female_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365280/SRR8365280_2.fastq.gz -o SRR8365280_HIP1_female_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365281/SRR8365281_2.fastq.gz -o SRR8365281_HIP1_female_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365282/SRR8365282_2.fastq.gz -o SRR8365282_HIP1_female_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365283/SRR8365283_2.fastq.gz -o SRR8365283_HIP1_female_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365284/SRR8365284_2.fastq.gz -o SRR8365284_HIP1_female_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365285/SRR8365285_2.fastq.gz -o SRR8365285_HIP2_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365286/SRR8365286_2.fastq.gz -o SRR8365286_HIP2_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365287/SRR8365287_2.fastq.gz -o SRR8365287_HIP2_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365288/SRR8365288_2.fastq.gz -o SRR8365288_HIP2_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365289/SRR8365289_2.fastq.gz -o SRR8365289_HIP2_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365290/SRR8365290_2.fastq.gz -o SRR8365290_HIP2_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365291/SRR8365291_2.fastq.gz -o SRR8365291_HIP2_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365292/SRR8365292_2.fastq.gz -o SRR8365292_HIP2_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365293/SRR8365293_2.fastq.gz -o SRR8365293_HIP3_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365294/SRR8365294_2.fastq.gz -o SRR8365294_HIP3_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365295/SRR8365295_2.fastq.gz -o SRR8365295_HIP3_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365296/SRR8365296_2.fastq.gz -o SRR8365296_HIP3_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365297/SRR8365297_2.fastq.gz -o SRR8365297_HIP2_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365298/SRR8365298_2.fastq.gz -o SRR8365298_HIP2_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365299/SRR8365299_2.fastq.gz -o SRR8365299_HIP3_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365300/SRR8365300_2.fastq.gz -o SRR8365300_HIP3_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365311/SRR8365311_2.fastq.gz -o SRR8365311_HIP1_female_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365312/SRR8365312_2.fastq.gz -o SRR8365312_HIP1_female_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365313/SRR8365313_2.fastq.gz -o SRR8365313_HIP1_female_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365314/SRR8365314_2.fastq.gz -o SRR8365314_HIP1_female_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365315/SRR8365315_2.fastq.gz -o SRR8365315_HIP1_female_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365316/SRR8365316_2.fastq.gz -o SRR8365316_HIP1_female_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365317/SRR8365317_2.fastq.gz -o SRR8365317_HIP1_female_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365318/SRR8365318_2.fastq.gz -o SRR8365318_HIP1_female_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365319/SRR8365319_2.fastq.gz -o SRR8365319_HIP1_female_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365320/SRR8365320_2.fastq.gz -o SRR8365320_HIP1_female_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365323/SRR8365323_2.fastq.gz -o SRR8365323_HIP3_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365332/SRR8365332_2.fastq.gz -o SRR8365332_HIP2_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365333/SRR8365333_2.fastq.gz -o SRR8365333_HIP2_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365334/SRR8365334_2.fastq.gz -o SRR8365334_HIP2_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365335/SRR8365335_2.fastq.gz -o SRR8365335_HIP2_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365336/SRR8365336_2.fastq.gz -o SRR8365336_HIP2_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365337/SRR8365337_2.fastq.gz -o SRR8365337_HIP2_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365338/SRR8365338_2.fastq.gz -o SRR8365338_HIP2_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365339/SRR8365339_2.fastq.gz -o SRR8365339_HIP2_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365340/SRR8365340_2.fastq.gz -o SRR8365340_HIP3_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365341/SRR8365341_2.fastq.gz -o SRR8365341_HIP3_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365342/SRR8365342_2.fastq.gz -o SRR8365342_HIP2_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365343/SRR8365343_2.fastq.gz -o SRR8365343_HIP2_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365344/SRR8365344_2.fastq.gz -o SRR8365344_HIP3_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365345/SRR8365345_2.fastq.gz -o SRR8365345_HIP3_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365346/SRR8365346_2.fastq.gz -o SRR8365346_HIP3_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365347/SRR8365347_2.fastq.gz -o SRR8365347_HIP3_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365349/SRR8365349_2.fastq.gz -o SRR8365349_HIP2_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365350/SRR8365350_2.fastq.gz -o SRR8365350_HIP2_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365351/SRR8365351_2.fastq.gz -o SRR8365351_HIP2_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365352/SRR8365352_2.fastq.gz -o SRR8365352_HIP2_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365353/SRR8365353_2.fastq.gz -o SRR8365353_HIP2_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365354/SRR8365354_2.fastq.gz -o SRR8365354_HIP2_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365355/SRR8365355_2.fastq.gz -o SRR8365355_HIP2_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365356/SRR8365356_2.fastq.gz -o SRR8365356_HIP2_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365357/SRR8365357_2.fastq.gz -o SRR8365357_HIP2_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365358/SRR8365358_2.fastq.gz -o SRR8365358_HIP2_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365359/SRR8365359_2.fastq.gz -o SRR8365359_HIP3_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365360/SRR8365360_2.fastq.gz -o SRR8365360_HIP3_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365361/SRR8365361_2.fastq.gz -o SRR8365361_HIP3_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365362/SRR8365362_2.fastq.gz -o SRR8365362_HIP3_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365363/SRR8365363_2.fastq.gz -o SRR8365363_HIP3_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365364/SRR8365364_2.fastq.gz -o SRR8365364_HIP3_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365365/SRR8365365_2.fastq.gz -o SRR8365365_HIP3_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365366/SRR8365366_2.fastq.gz -o SRR8365366_HIP3_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365367/SRR8365367_2.fastq.gz -o SRR8365367_HIP3_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365368/SRR8365368_2.fastq.gz -o SRR8365368_HIP3_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365369/SRR8365369_2.fastq.gz -o SRR8365369_HIP3_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365370/SRR8365370_2.fastq.gz -o SRR8365370_HIP2_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365371/SRR8365371_2.fastq.gz -o SRR8365371_HIP2_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365372/SRR8365372_2.fastq.gz -o SRR8365372_HIP3_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365373/SRR8365373_2.fastq.gz -o SRR8365373_HIP2_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365374/SRR8365374_2.fastq.gz -o SRR8365374_HIP2_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365375/SRR8365375_2.fastq.gz -o SRR8365375_HIP2_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365376/SRR8365376_2.fastq.gz -o SRR8365376_HIP2_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365377/SRR8365377_2.fastq.gz -o SRR8365377_HIP2_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365378/SRR8365378_2.fastq.gz -o SRR8365378_HIP2_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365379/SRR8365379_2.fastq.gz -o SRR8365379_HIP2_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365380/SRR8365380_2.fastq.gz -o SRR8365380_HIP2_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365381/SRR8365381_2.fastq.gz -o SRR8365381_HIP3_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365382/SRR8365382_2.fastq.gz -o SRR8365382_HIP3_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365383/SRR8365383_2.fastq.gz -o SRR8365383_HIP3_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365384/SRR8365384_2.fastq.gz -o SRR8365384_HIP3_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365385/SRR8365385_2.fastq.gz -o SRR8365385_HIP3_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365386/SRR8365386_2.fastq.gz -o SRR8365386_HIP3_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365387/SRR8365387_2.fastq.gz -o SRR8365387_HIP3_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365388/SRR8365388_2.fastq.gz -o SRR8365388_HIP3_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365389/SRR8365389_2.fastq.gz -o SRR8365389_HIP3_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365390/SRR8365390_2.fastq.gz -o SRR8365390_HIP3_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365391/SRR8365391_2.fastq.gz -o SRR8365391_HIP1_female_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365392/SRR8365392_2.fastq.gz -o SRR8365392_HIP1_female_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365393/SRR8365393_2.fastq.gz -o SRR8365393_HIP1_female_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365394/SRR8365394_2.fastq.gz -o SRR8365394_HIP1_female_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365395/SRR8365395_2.fastq.gz -o SRR8365395_HIP1_female_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365396/SRR8365396_2.fastq.gz -o SRR8365396_HIP1_female_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365397/SRR8365397_2.fastq.gz -o SRR8365397_HIP1_female_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365398/SRR8365398_2.fastq.gz -o SRR8365398_HIP1_female_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365399/SRR8365399_2.fastq.gz -o SRR8365399_HIP1_female_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365400/SRR8365400_2.fastq.gz -o SRR8365400_HIP1_female_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365401/SRR8365401_2.fastq.gz -o SRR8365401_HIP3_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365402/SRR8365402_2.fastq.gz -o SRR8365402_HIP3_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365413/SRR8365413_2.fastq.gz -o SRR8365413_HIP3_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365414/SRR8365414_2.fastq.gz -o SRR8365414_HIP3_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365415/SRR8365415_2.fastq.gz -o SRR8365415_HIP3_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365416/SRR8365416_2.fastq.gz -o SRR8365416_HIP3_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365417/SRR8365417_2.fastq.gz -o SRR8365417_HIP3_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365425/SRR8365425_2.fastq.gz -o SRR8365425_HIP3_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365426/SRR8365426_2.fastq.gz -o SRR8365426_HIP3_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365427/SRR8365427_2.fastq.gz -o SRR8365427_HIP3_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365428/SRR8365428_2.fastq.gz -o SRR8365428_HIP1_female_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365429/SRR8365429_2.fastq.gz -o SRR8365429_HIP1_female_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365430/SRR8365430_2.fastq.gz -o SRR8365430_HIP1_female_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365431/SRR8365431_2.fastq.gz -o SRR8365431_HIP1_female_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365432/SRR8365432_2.fastq.gz -o SRR8365432_HIP1_female_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365433/SRR8365433_2.fastq.gz -o SRR8365433_HIP1_female_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365442/SRR8365442_2.fastq.gz -o SRR8365442_HIP3_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365445/SRR8365445_2.fastq.gz -o SRR8365445_HIP3_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365451/SRR8365451_2.fastq.gz -o SRR8365451_HIP2_male_IgG1_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365452/SRR8365452_2.fastq.gz -o SRR8365452_HIP2_male_IgG2_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/003/SRR8365453/SRR8365453_2.fastq.gz -o SRR8365453_HIP2_male_IgG3_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/004/SRR8365454/SRR8365454_2.fastq.gz -o SRR8365454_HIP2_male_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/005/SRR8365455/SRR8365455_2.fastq.gz -o SRR8365455_HIP2_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365466/SRR8365466_2.fastq.gz -o SRR8365466_HIP3_male_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/002/SRR8365472/SRR8365472_2.fastq.gz -o SRR8365472_HIP2_male_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365476/SRR8365476_2.fastq.gz -o SRR8365476_HIP2_male_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365477/SRR8365477_2.fastq.gz -o SRR8365477_HIP2_male_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/008/SRR8365478/SRR8365478_2.fastq.gz -o SRR8365478_HIP1_female_IgL_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/009/SRR8365479/SRR8365479_2.fastq.gz -o SRR8365479_HIP1_female_IgK_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/000/SRR8365480/SRR8365480_2.fastq.gz -o SRR8365480_HIP1_female_IgA_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/001/SRR8365481/SRR8365481_2.fastq.gz -o SRR8365481_HIP1_female_IgM_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/006/SRR8365486/SRR8365486_2.fastq.gz -o SRR8365486_HIP1_female_IgG4_R2.fastq.gz
curl -L ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR836/007/SRR8365487/SRR8365487_2.fastq.gz -o SRR8365487_HIP1_female_IgG3_R2.fastq.gz
</code></pre></div> </details> <p>The project contains 136 FASTQ file pairs. For the purpose of this tutorial we assume that all fastq files are stored in <code>fastq/</code> folder. Each file name encodes the information about donor, sex and Ig chain and isotype. For example for the first file from above listing: </p> <ul> <li><code>SRR8365277</code> - sample id</li> <li><code>HIP1</code> - patient id</li> <li><code>female</code> - patient sex</li> <li><code>IgG1</code> - Ig isotype</li> </ul> <p>The structure of sequences is shown on the picture bellow. </p> <figure> <p><a class="glightbox" href="ABHelix.svg"><img alt="ABHelix.svg" src="ABHelix.svg"/></a></p> </figure> <h2 id="upstream-analysis">Upstream analysis</h2> <p>The easiest way to obtain results from this type of data is to use <code>mixcr analyze amplicon</code> command in the following manner:</p> <p><div class="highlight"><pre><span></span><code>&gt; mixcr analyze amplicon <span class="se">\ </span>
    --species hsa <span class="se">\</span>
    --starting-material rna <span class="se">\</span>
    --receptor-type bcr <span class="se">\</span>
    --5-end no-v-primers <span class="se">\</span>
    --3-end j-primers <span class="se">\</span>
    --adapters adapters-present <span class="se">\</span>
    fastq/SRR8365277_HIP1_female_IgG1_R1.fastq.gz fastq/SRR8365277_HIP1_female_IgG1_R2.fastq.gz <span class="se">\</span>
    result/SRR8365277_HIP1_female_IgG1_
</code></pre></div> Arguments explained:</p> <dl> <dt><code>--species</code></dt> <dd>is set to <code>hsa</code> for <em>Homo Sapience</em></dd> <dt><code>--starting-material</code></dt> <dd><code>rna</code> It affects the choice of V gene region which will be used as target in <a href="../../reference/mixcr-align/"><code>align</code></a> step (<code>vParameters.geneFeatureToAlign</code>, see <a href="../../reference/mixcr-align/"><code>align</code> documentation</a>). By specifying <code>rna</code> as starting material, <code>VTranscriptWithout5UTRWithP</code> will be used as <code>geneFeatureToAlign</code> for V segment.</dd> <dt><code>--receptor-type</code></dt> <dd><code>bcr</code>. It affects the choice of underlying alignment algorithms. Due to somatic hypermutations and long indels MiXCR uses a fundamentally different algorithm for BCRs.</dd> <dt><code>--5-end</code></dt> <dd>is set to <code>no-v-primers</code> because samples were prepared using primers located in the leader region. This leads to a global alignment on the left bound of V gene.</dd> <dt><code>--3-end</code></dt> <dd>is set to <code>j-primers</code>, since the J multiplex primers were used for library preparation. This choice leads to a local alignment on the right bound of J gene.(<code>adapters-present</code> required)</dd> <dt><code>--adapers</code></dt> <dd>is set to <code>adapters-present</code> because primers sequences are present in the data and has not been cut prior to. Together with <code>--3-end j-primers</code> it leads to a local alignment on the right bound of J gene.</dd> </dl> <p>One may use the script bellow to process all samples together (see <a href="https://www.gnu.org/software/parallel/">GNU Parallel</a>):</p> <div class="highlight"><pre><span></span><code>&gt; ls fastq/*R1* <span class="p">|</span> parallel -j <span class="m">4</span> <span class="se">\</span>
    <span class="s1">'mixcr analyze amplicon \</span>
<span class="s1">          --species hsa \</span>
<span class="s1">          --starting-material rna \</span>
<span class="s1">          --receptor-type bcr \</span>
<span class="s1">          --5-end no-v-primers \</span>
<span class="s1">          --3-end j-primers \</span>
<span class="s1">          --adapters adapters-present \</span>
<span class="s1">          {} {=s:R1:R2:=} \</span>
<span class="s1">          {=s:.*/:results/:;s:_R.*::=}'</span>
</code></pre></div> <h3 id="under-the-hood-pipeline">Under the hood pipeline:</h3> <p>Under the hood the command above actually executes the following pipeline:</p> <h4 id="align"><code>align</code></h4> <p>Alignment of raw sequencing reads against reference database of V-, D-, J- and C- gene segments.</p> <div class="highlight"><pre><span></span><code><span class="c1"># align raw reads</span>
&gt; mixcr align <span class="se">\</span>
    -s hsa <span class="se">\</span>
    -p kAligner2 <span class="se">\</span>
    -OvParameters.geneFeatureToAlign<span class="o">=</span><span class="s2">"VTranscriptWithout5UTRWithP"</span> <span class="se">\</span>
    -OvParameters.parameters.floatingLeftBound<span class="o">=</span><span class="nb">false</span> <span class="se">\</span>
    -OjParameters.parameters.floatingRightBound<span class="o">=</span><span class="nb">true</span> <span class="se">\</span>
    -OcParameters.parameters.floatingRightBound<span class="o">=</span><span class="nb">false</span> <span class="se">\</span>
    --report result/SRR8365277_HIP1_female_IgG1.report <span class="se">\</span>
     fastq/SRR8365277_HIP1_female_IgG1_R1.fastq.gz fastq/SRR8365277_HIP1_female_IgG1_R2.fastq.gz <span class="se">\</span>
     result/SRR8365277_HIP1_female_IgG1.vdjca
</code></pre></div> <p>Option <code>--report</code> is specified here explicitly.</p> <dl> <dt><code>-p</code></dt> <dd><code>kAligner2</code> specifies the specific BCR aligner.</dd> <dt><code>-OvParameters.geneFeatureToAlign="VTranscriptWithout5UTRWithP"</code></dt> <dd>Sets a V gene feature to align. Check <a href="../../reference/ref-gene-features/">gene features</a> for more info.</dd> <dt><code>-OvParameters.parameters.floatingLeftBound=false</code></dt> <dd>Results in a global alignment algorithm for V gene left bound. </dd> <dt><code>-OjParameters.parameters.floatingRightBound=true</code></dt> <dd>Results in a local alignment algorithm for J gene right bound.</dd> <dt><code>-OcParameters.parameters.floatingRightBound=false</code></dt> <dd>Results in a global alignment algorithm for C gene right bound.</dd> </dl> <h4 id="assemble"><code>assemble</code></h4> <p>Assembles alignments into clonotypes and applies several layers of errors correction(ex. quality-awared correction for sequencing errors, clustering to correct for PCR errors). Check <a href="../../reference/mixcr-assemble/"><code>mixcr assemble</code></a> for more information.</p> <div class="highlight"><pre><span></span><code><span class="c1"># assemble clonotypes</span>
&gt; mixcr assemble <span class="se">\</span>
    -OseparateByV<span class="o">=</span><span class="nb">true</span> <span class="se">\</span>
    result/SRR8365277_HIP1_female_IgG1.vdjca <span class="se">\</span>
    result/SRR8365277_HIP1_female_IgG1.clns
</code></pre></div> <dl> <dt><code>-OseparateByV=true</code></dt> <dd>Separate clones with same <code>CDR3</code> sequence but different V-genes. This option is important for BCR data due to hypermutations in V region, but it shouldn't be used unless no V gene primer sequences present in the data.</dd> </dl> <h4 id="export"><code>export</code></h4> <p>Exports clonotypes from .clns file into human-readable tables.</p> <p><div class="highlight"><pre><span></span><code><span class="c1"># export to tsv</span>
&gt; mixcr exportClones <span class="se">\</span>
    -p full <span class="se">\</span>
    -c IGH
    result/SRR8365277_HIP1_female_IgG1.clns <span class="se">\</span>
    result/SRR8365277_HIP1_female_IgG1.clonotypes.IGHtxt

&gt; mixcr exportClones <span class="se">\</span>
    -p full <span class="se">\</span>
    -c IGK
    result/SRR8365277_HIP1_female_IgG1.clns <span class="se">\</span>
    result/SRR8365277_HIP1_female_IgG1.clonotypes.IGK.txt

&gt; mixcr exportClones <span class="se">\</span>
    -p full <span class="se">\</span>
    -c IGL
    result/SRR8365277_HIP1_female_IgG1.clns <span class="se">\</span>
    result/SRR8365277_HIP1_female_IgG1.clonotypes.IGL.txt
</code></pre></div> Here <code>-p full</code> defines the full preset of common export columns. Check <a href="../../reference/mixcr-export/"><code>mixcr export</code></a> for more information.</p> <dl> <dt><code>-с &lt;chain&gt;</code></dt> <dd>defines a specific chain to be exported. By default <code>mixcr analyze</code> exports all chains in separate files.</dd> </dl> <p>By the end of upstream analysis you should have the following set of output files:</p> <div class="highlight"><pre><span></span><code><span class="c1"># human-readable reports </span>
SRR8365277_HIP1_female_IgG1.report

<span class="c1"># raw alignments (highly compressed binary file)</span>
SRR8365277_HIP1_female_IgG1.vdjca

<span class="c1"># IGH, IGK and IGL clonotypes (highly compressed binary file)</span>
SRR8365277_HIP1_female_IgG1.clns

<span class="c1"># IGH, IGK and IGL clonotypes exported in tab-delimited txt</span>
SRR8365277_HIP1_female_IgG1.clonotypes.IGH.txt
SRR8365277_HIP1_female_IgG1.clonotypes.IGK.txt
SRR8365277_HIP1_female_IgG1.clonotypes.IGL.txt
</code></pre></div> <h2 id="quality-control">Quality control</h2> <p>Now when we have processed all samples, we can proceed to quality control. First thing that we want to look at is the alignment quality. This can be easily done with the following command:</p> <div class="highlight"><pre><span></span><code>&gt; mixcr exportQc align <span class="se">\</span>
    result/*.vdjca <span class="se">\</span>
    alignQc.pdf
</code></pre></div> <figure> <p><a class="glightbox" href="alignQc.svg"><img alt="alignQc.svg" src="alignQc.svg"/></a></p> </figure> <p>From the plot above we see that most samples have high percentage of successfully aligned reads (above 90%). What we can also tell is that from all the samples those that come from HIP1_female_IgG4 have a lower alignment rate:</p> <ul> <li>SRR8365486_HIP1_female_IgG4</li> <li>SRR8365430_HIP1_female_IgG4</li> <li>SRR8365394_HIP1_female_IgG4</li> <li>SRR8365317_HIP1_female_IgG4</li> <li>SRR8365280_HIP1_female_IgG4</li> </ul> <p>That suggest a potential issue during sample preparation. Note that HIP3_male_IgG4 and HIP3_male_IgG3 sample also tend to have a lower alignment rate.</p> <p>Let's look at the same plot, but instead of percentages of reads we will plot an absolute number of reads.</p> <div class="highlight"><pre><span></span><code>&gt; mixcr exportQc align <span class="se">\</span>
    --absolute-values <span class="se">\</span>
    test_run/*.vdjca  <span class="se">\</span>
    alignQcAbsolute.svg
</code></pre></div> <figure> <p><a class="glightbox" href="alignQcAbsolute.svg"><img alt="alignQcAbsolute.svg" src="alignQcAbsolute.svg"/></a></p> </figure> <p>Now we see, that nearly all bad samples have a significantly lower total reads count. Next, lets take a closer look at SRR8365280_HIP1_female_IgG4, because it has a lot of reads compared to other low quality samples but most of them lack Ig sequences. By default, MiXCR removed non target reads during alignment. We will realign this sample using options <code>-OallowPartialAlignments=true</code> and <code>-OallowNoCDR3PartAlignments=true</code> to preserve partially aligned reads. We will also save not aligned reads (<code>--not-aligned-R1</code>, <code>--not-aligned-R2</code>) to separate FASTQ files for manual inspection. See <a href="../../reference/mixcr-align/"><code>mixcr align</code></a> for more details.</p> <p>Bellow is the complete command:</p> <div class="highlight"><pre><span></span><code>&gt; mixcr align <span class="se">\</span>
    -s hsa <span class="se">\</span>
    -p kAligner2 <span class="se">\</span>
    -OvParameters.geneFeatureToAlign<span class="o">=</span><span class="s2">"VTranscriptWithout5UTRWithP"</span> <span class="se">\</span>
    -OvParameters.parameters.floatingLeftBound<span class="o">=</span><span class="nb">false</span> <span class="se">\</span>
    -OjParameters.parameters.floatingRightBound<span class="o">=</span><span class="nb">true</span> <span class="se">\</span>
    -OcParameters.parameters.floatingRightBound<span class="o">=</span><span class="nb">false</span> <span class="se">\</span>
    -OallowPartialAlignments<span class="o">=</span><span class="nb">true</span> <span class="se">\</span>
    -OallowNoCDR3PartAlignments<span class="o">=</span><span class="nb">true</span> <span class="se">\</span>
    --not-aligned-R1 SRR8365280_HIP1_female_IgG4_notAligned_R1.fastq <span class="se">\</span>
    --not-aligned-R2 SRR8365280_HIP1_female_IgG4_notAligned_R2.fastq <span class="se">\</span>
    -r SRR8365280_HIP1_female_IgG4_debug.report <span class="se">\</span>
     fastq/SRR8365280_HIP1_female_IgG4_R1.fastq.gz fastq/SRR8365280_HIP1_female_IgG4_R2.fastq.gz <span class="se">\</span>
     SRR8365280_HIP1_female_IgG4_debug.vdjca
</code></pre></div> <p>Resulting <code>SRR8365280_HIP1_female_IgG4_notAligned_R1.fastq</code> and <code>SRR8365280_HIP1_female_IgG4_notAligned_R1.fastq</code> files can be manually expected. A brief <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> search revealed that a lot of reads align with PhiX, which is regularly used as a DNA sequencing control in Illumina protocol. Bellow you can see a few reads from <code>SRR8365280_HIP1_female_IgG4_notAligned_R2.fastq</code>.</p> <div class="highlight"><pre><span></span><code>@SRR8365280.116 GWZHISEQ01:570:HYJYKBCXX:1:1101:12039:44077/2
GAGAGATGAGATTGAGGCTGGGAAAAGTTACTGTAGCCGACGTTTTGGCGGCGCAACCTGTGACGACAAATCTGCTCAAATTTATGCGCGCTTCGATAAAAATGATTGGCGTATCCAACCTGCAGAGTTTTATCGCTTCCATGACGCAGAAGTTAACACTTTCGGATATTTCTGATGAGTCGAAAAATTATCTTGATAAAGCAGGAATTACTACTGCTTGTTTACGAATTAAATCGAAGTGGACTGCTGGC
+
DDBDD&lt;<span class="nv">GHHIFHIHIIIEHIGGIIIIIIGHIIIICHIIIIIHEHHHHDHIICHHHIIIIIIIIIIIGDHIIFHIIIIIGIIIIIHIHHIHIIHIIHHIIIIHIIIIIIIIIIIIIGIIIIHIIIIEHHIIHHHGIIIIIIIIIIIIIHIIIHIHHIIIIIIHIIIII</span><span class="o">=</span>FHHIFHCGHG?HIIIIGIIIIEHCFHHII?GHH.FHEHIIHHHH-BGHHHHGFHIIIHI.5B6-8@F-B@HHHHIIIIGFE?-
@SRR8365280.117 GWZHISEQ01:570:HYJYKBCXX:1:1101:13143:44082/2
ACGATGAGGGACATAAAAAGTAAAAATGTCTACAGTAGAGTCAATAGCAAGGCCACGACGCAATGGAGAAAGACGGAGAGCGCCAACGGCGTCCATCTCGAAGGAGTCGCCAGCGATAACCGGAGTAGTTGAAATGGTAATAAGACGACCAATCTGACCAGCAAGGAAGCCAAGATGGGAAAGGTCATGCGGCATACGCTCGGCGCCAGTTTGAATATTAGACATAATTTATCCTCAAGTAAGGGGCCGAA
+
DDDDBIIIIIHHGHHIHHHIIIFIGIHHIIHHIIIIIHHIHIIIIEHHHGIIIIIIIIIHHIIIIIIHIIIFHIHIIIIIHGHIIIIHIIIIIHIHIIIIIDGHHHHIGIIHIIIIIIIIIGHIHGHHHHHIHIIEHIIII@HHIIIIIIIHIHIIHHIHIHIIIIIHIIII.BEGHGII?FHGHFHHHCDHIIIIHHCEHHIHIGIIIAHGEEEHH?FH?AFFHFHHHHH?FCH--BHHHHHE@,,&gt;??H
@SRR8365280.118 GWZHISEQ01:570:HYJYKBCXX:1:1101:9938:44446/2
AGAAGAAAACGTGCGTCAAAAATTACGTGCAGAAGGAGTGATGTAATGTCTAAAGGTAAAAAACGTTCTGGCGCTCGCCCTGGTCGTCCGCAGCCGTTGCGAGGTACTAAAGGCAAGCGTAAAGGCGCTCGTCTTTGGTATGTAGGTGGTCAACAATTTTAATTGCAGGGGCTTCGGCCCCTTACTTGAGGATAAATTATGTCTAATATTCAAACTGGCGCCGAGCGTATGCCGCATGACCTTTCCCATCT
+
DDDDDIIGIIIIIIHGHHHHHHHIIIIHIF?HHIII?FHIIHIIIEHFHHIIIIIIIIHIIIIIIHIIIHIIIIIIIIIIIIIIIIIGIIIIIIIIHHHIHHHHIIIIHIIIIIIIHIIIHIIHIGIICGHCHHIHHGEEEEHIIIIIIIIIIIGHIIIIIIIIHIIICE&lt;CHIIHHIIHIHIIIIFHIIIIIIHHEGEHHAFEECHIGHIIIIIG?E?EEH?CHGIGHII?GHIDDHHHIIIHEHEFB-A
</code></pre></div> <p>Now lets look inside <code>SRR8365280_HIP1_female_IgG4_debug.vdjca</code> file witch now has partially aligned reads in it. That can be easily achieved using <a href="../../reference/mixcr-exportPretty/#raw-alignments"><code>mixcr exportAlignmentsPretty</code></a>.</p> <p>The function bellow will generate a <code>.txt</code> human-readable file with alignments. We use parameter <code>--skip 1000</code> to skip first 1000 reads, as first reads usually have bad quality, and <code>--limit 100</code> will export only 100 alignments.</p> <div class="highlight"><pre><span></span><code>&gt; mixcr exportAlignmentsPretty <span class="se">\</span>
    --skip <span class="m">1000</span> <span class="se">\</span>
    --limit <span class="m">100</span> <span class="se">\</span>
    SRR8365280_HIP1_female_IgG4_debug.vdjca <span class="se">\</span>
    SRR8365280_HIP1_female_IgG4_debug.alignments.txt
</code></pre></div> <p>If we look in the file we can spot how some troubled alignments look like. Bellow you can see one alignment from that file. First read aligns with a good score to IGHV1-69D. But its pair consists of low-quality nucleotides and can't be aligned against ahy IGHJ.</p> <p>Further analysis might help reveal where such low quality reads come from.</p> <div class="highlight"><pre><span></span><code>&gt;&gt;&gt; Read ids: <span class="m">12846</span>


                                                L1&gt;&lt;L2     L2&gt;&lt;FR1
                _ T  R  F  L  F  V  V  A  A  A  T  G  V  Q  S  Q  V  Q  L  V  Q  S  G  P  E  V
     Quality    <span class="m">76767888777877765777666775777788877737576737677788888767777778357777767777667877</span>
     Target0  <span class="m">0</span> CACTAGGTTCCTCTTTGTGGTGGCAGCAGCTACAGGTGTCCAGTCCCAGGTGCAGCTGGTGCAGTCTGGGCCTGAGGTGA <span class="m">79</span>  Score
IGHV1-69D*00 <span class="m">11</span> cTGGaggttcctctttgtggtggcagcagctacaggtgtccagtcccaggtgcagctggtgcagtctgggGctgaggtga <span class="m">90</span>  <span class="m">1901</span>

                                                     FR1&gt;&lt;CDR1              CDR1&gt;&lt;FR2
                K  K  P  G  S  S  V  K  V  S  C  E  A  S  G  G  T  F  S  N  F  A  V  N  W  V  R
     Quality    <span class="m">87888877888677675673763577787777777786767777776667878888888773577677878867775677</span>
     Target0 <span class="m">80</span> AGAAGCCTGGGTCGTCGGTGAAGGTCTCCTGCGAGGCTTCTGGAGGCACCTTCAGTAACTTTGCTGTCAACTGGGTGCGA <span class="m">159</span>  Score
IGHV1-69D*00 <span class="m">91</span> agaagcctgggtcCtcggtgaaggtctcctgcAaggcttctggaggcaccttcagCaGctAtgctAtcaGctgggtgcga <span class="m">170</span>  <span class="m">1901</span>

                                                 FR2&gt;&lt;CDR2              CDR2&gt;&lt;FR3
                  Q  A  P  G  Q  G  L  E  W  V  G  G  I  I  P  L  F  N  V  A  K  Y  A  Q  K  F  E
     Quality     <span class="m">77787778865577867376625733535577887777787767786772253677767777677777787777766667</span>
     Target0 <span class="m">160</span> CAGGCCCCTGGACAAGGGCTTGAGTGGGTGGGAGGGATCATCCCTCTATTTAATGTGGCAAAGTACGCACAGAAGTTCGA <span class="m">239</span>  Score
IGHV1-69D*00 <span class="m">171</span> caggcccctggacaagggcttgagtggAtgggagggatcatccctAtCtttGGtACAgcaaaCtacgcacagaagttcCa <span class="m">250</span>  <span class="m">1901</span>


                   G  R  V _
     Quality     <span class="m">76677566676</span>
     Target0 <span class="m">240</span> GGGCAGAGTCA <span class="m">250</span>  Score
IGHV1-69D*00 <span class="m">251</span> gggcagagtca <span class="m">261</span>  <span class="m">1901</span>


Quality   <span class="m">26426422222222222562442222424424255222225225222225224226522255252272242277525257</span>
Target1 <span class="m">0</span> GGTTGGTTGGCGGGCTCATCCCGCTATTTAATGTGGGGATGTACGCACTGAGGTACGTTGGCAGGGTCCCGGTTGTCGCG <span class="m">79</span>  Score


Quality    <span class="m">72737277523535532353332333335333222235526353536262252522222577533555352225673735</span>
Target1 <span class="m">80</span> GACGATTCAGTGGGCTGTGCTTACGTAGACATTGCCCGCTTGCGATCTGCCGCCAGCGCCGTGTATTACTGTGCGGCTTC <span class="m">159</span>  Score


Quality     <span class="m">53333223353333655367555355333325223333535353355335367777733353335333333533533533</span>
Target1 <span class="m">160</span> AGGTGGCGACGTCCTGTGATATGACTACAAGGCCCCCTAAGGGGATGAACTATGGGGGCAATCGACAATAGTGACCGTCT <span class="m">239</span>  Score


Quality     <span class="m">33533233333</span>
Target1 <span class="m">240</span> ACTAAGTTTCG <span class="m">250</span>  Score
</code></pre></div> <p>Finally, lets check chain usage among all samples.</p> <div class="highlight"><pre><span></span><code>&gt; mixcr exportQc chainUsage <span class="se">\</span>
    test_run/*.vdjca <span class="se">\</span>
    chainUsage.pdf
</code></pre></div> <p>Bellow is plot visualizing chain usage distribution among all samples. The plot suggests that this data has a low cross-contamination level, since almost all clones from IgG1, IgG2,IgG3,IgG4,IgM and IgA samples are IGH, and samples with IgK and IgL mostly consist of IGK and IGL clones.</p> <figure> <p><a class="glightbox" href="chainUsage.svg"><img alt="chainUsage.svg" src="chainUsage.svg"/></a></p> </figure> <h2 id="advanced-settings">Advanced settings</h2> <p>By default <code>mixcr assemble</code> assembles clones by <code>CDR3</code> sequence. But in case of BCR data, due to hypermutations in V region, we usually want to extend that assemble feature so clones with different V gene sequences will be separated regardless of the same V-gene name. This is only possible if ore reads long enough to cover most of the V gene and if we a sure, that primer sequences will not introduce any bias.</p> <p>ABhelix library preparation protocol is designed in such a way that forward primers are located in the V gene leader sequence. Thus, when we assemble clones we can actually use that part of V gene. The easiest way to utilize that "extra" sequence is to pass an extended <a href="../../reference/mixcr-assemble/">assembling feature</a> to <code>mixcr analyze amplicon</code> command.</p> <div class="highlight"><pre><span></span><code>&gt; mixcr analyze amplicon <span class="se">\</span>
    --species hsa <span class="se">\</span>
    --starting-material rna <span class="se">\</span>
    --receptor-type bcr <span class="se">\</span>
    --5-end no-v-primers <span class="se">\</span>
    --3-end j-primers <span class="se">\</span>
    --adapters adapters-present <span class="se">\</span>
    --assemble <span class="s2">"-OassemblingFeatures={FR1Begin:CDR3End}"</span>
    fastq/SRR8365277_HIP1_female_IgG1_R1.fastq.gz fastq/SRR8365277_HIP1_female_IgG1_R2.fastq.gz <span class="se">\</span>
    result/SRR8365277_HIP1_female_IgG1_
</code></pre></div> <p>Under the hood <code>-OassemblingFeatures={FR1Begin:CDR3End}</code> will be passed to <a href="../../reference/mixcr-assemble/"><code>mixcr asemble</code></a>. </p> <hr/> <div class="md-source-file"> <small> Last update: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago"><span class="timeago" datetime="2022-08-28T19:13:52+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date">2022-08-28</span> <br/> Created: <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago"><span class="timeago" datetime="2022-08-28T15:04:50+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date">2022-08-28</span> </small> </div> </article> <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script> </div> </div> <a class="md-top md-icon" data-md-component="top" hidden="" href="#"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg> Back to top </a> </main> <footer class="md-footer"> <nav aria-label="Footer" class="md-footer__inner md-grid"> <a aria-label="Previous: BioMed2" class="md-footer__link md-footer__link--prev" href="../biomed2-bcr/" rel="prev"> <div class="md-footer__button md-icon"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg> </div> <div class="md-footer__title"> <div class="md-ellipsis"> <span class="md-footer__direction"> Previous </span> BioMed2 </div> </div> </a> <a aria-label="Next: SMARTer Mouse TCR a/b Profiling Kit" class="md-footer__link md-footer__link--next" href="../takara-mmu-tcr/" rel="next"> <div class="md-footer__title"> <div class="md-ellipsis"> <span class="md-footer__direction"> Next </span> SMARTer Mouse TCR a/b Profiling Kit </div> </div> <div class="md-footer__button md-icon"> <svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"></path></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class="md-copyright"> Made with <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank"> Material for MkDocs </a> </div> </div> </div> </footer> </div> <div class="md-dialog" data-md-component="dialog"> <div class="md-dialog__inner md-typeset"></div> </div> <script id="__config" type="application/json">{"base": "../../..", "features": ["navigation.instant", "navigation.tabs", "navigation.top", "navigation.sections", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "content.tabs.link"], "search": "../../../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search by name or category", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src="../../../assets/javascripts/bundle.6c7ad80a.min.js"></script> <script src="../../../js/timeago.min.js"></script> <script src="../../../js/timeago_mkdocs_material.js"></script> <script src="../../../javascripts/svg.js"></script> <script src="../../../javascripts/onload.js"></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script> <script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "width": "100%", "height": "auto", "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom"});})</script></body> </html>